<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml" lang="" xml:lang="">

<head>
  <meta charset="utf-8"/>
  <meta name="viewport" content="width=device-width, initial-scale=1"/>
  <meta http-equiv="X-UA-Compatible" content="IE=Edge,chrome=1"/>
  <meta name="generator" content="distill" />

  <style type="text/css">
  /* Hide doc at startup (prevent jankiness while JS renders/transforms) */
  body {
    visibility: hidden;
  }
  </style>

 <!--radix_placeholder_import_source-->
 <!--/radix_placeholder_import_source-->

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  { color: #00769e; background-color: #f1f3f5; }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span { color: #00769e; } /* Normal */
code span.al { color: #ad0000; } /* Alert */
code span.an { color: #5e5e5e; } /* Annotation */
code span.at { color: #657422; } /* Attribute */
code span.bn { color: #ad0000; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #00769e; } /* ControlFlow */
code span.ch { color: #20794d; } /* Char */
code span.cn { color: #8f5902; } /* Constant */
code span.co { color: #5e5e5e; } /* Comment */
code span.cv { color: #5e5e5e; font-style: italic; } /* CommentVar */
code span.do { color: #5e5e5e; font-style: italic; } /* Documentation */
code span.dt { color: #ad0000; } /* DataType */
code span.dv { color: #ad0000; } /* DecVal */
code span.er { color: #ad0000; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #ad0000; } /* Float */
code span.fu { color: #4758ab; } /* Function */
code span.im { } /* Import */
code span.in { color: #5e5e5e; } /* Information */
code span.kw { color: #00769e; } /* Keyword */
code span.op { color: #5e5e5e; } /* Operator */
code span.ot { color: #00769e; } /* Other */
code span.pp { color: #ad0000; } /* Preprocessor */
code span.sc { color: #5e5e5e; } /* SpecialChar */
code span.ss { color: #20794d; } /* SpecialString */
code span.st { color: #20794d; } /* String */
code span.va { color: #111111; } /* Variable */
code span.vs { color: #20794d; } /* VerbatimString */
code span.wa { color: #5e5e5e; font-style: italic; } /* Warning */
</style>


  <!--radix_placeholder_meta_tags-->
  <title>Rohan Alexander: Toronto Workshop on Reproducibility</title>

  <meta property="description" itemprop="description" content="An annual workshop focused on reproducibility in data science and statistics. First held 25-26 February 2021, and again 23-25 February 2022. Free and hosted via Zoom. Jointly hosted by CANSSI Ontario and the Data Sciences Institute. Supported by the Faculty of Information and the Department of Statistical Sciences."/>

  <link rel="icon" type="image/svg+xml" href="images/user-astronaut-solid.svg"/>


  <!--  https://developers.facebook.com/docs/sharing/webmasters#markup -->
  <meta property="og:title" content="Rohan Alexander: Toronto Workshop on Reproducibility"/>
  <meta property="og:type" content="article"/>
  <meta property="og:description" content="An annual workshop focused on reproducibility in data science and statistics. First held 25-26 February 2021, and again 23-25 February 2022. Free and hosted via Zoom. Jointly hosted by CANSSI Ontario and the Data Sciences Institute. Supported by the Faculty of Information and the Department of Statistical Sciences."/>
  <meta property="og:locale" content="en_US"/>
  <meta property="og:site_name" content="Rohan Alexander"/>

  <!--  https://dev.twitter.com/cards/types/summary -->
  <meta property="twitter:card" content="summary"/>
  <meta property="twitter:title" content="Rohan Alexander: Toronto Workshop on Reproducibility"/>
  <meta property="twitter:description" content="An annual workshop focused on reproducibility in data science and statistics. First held 25-26 February 2021, and again 23-25 February 2022. Free and hosted via Zoom. Jointly hosted by CANSSI Ontario and the Data Sciences Institute. Supported by the Faculty of Information and the Department of Statistical Sciences."/>
  <meta property="twitter:site" content="@RohanAlexander"/>

  <!--/radix_placeholder_meta_tags-->
  <!--radix_placeholder_rmarkdown_metadata-->

  <script type="text/json" id="radix-rmarkdown-metadata">
  {"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["title","description","output"]}},"value":[{"type":"character","attributes":{},"value":["Toronto Workshop on Reproducibility"]},{"type":"character","attributes":{},"value":["An annual workshop focused on reproducibility in data science and statistics. First held 25-26 February 2021, and again 23-25 February 2022. Free and hosted via Zoom. Jointly hosted by CANSSI Ontario and the Data Sciences Institute. Supported by the Faculty of Information and the Department of Statistical Sciences.\n"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["distill::distill_article"]}},"value":[{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["toc","toc_depth"]}},"value":[{"type":"logical","attributes":{},"value":[true]},{"type":"integer","attributes":{},"value":[2]}]}]}]}
  </script>
  <!--/radix_placeholder_rmarkdown_metadata-->
  <!--radix_placeholder_navigation_in_header-->
  <meta name="distill:offset" content=""/>

  <script type="application/javascript">

    window.headroom_prevent_pin = false;

    window.document.addEventListener("DOMContentLoaded", function (event) {

      // initialize headroom for banner
      var header = $('header').get(0);
      var headerHeight = header.offsetHeight;
      var headroom = new Headroom(header, {
        tolerance: 5,
        onPin : function() {
          if (window.headroom_prevent_pin) {
            window.headroom_prevent_pin = false;
            headroom.unpin();
          }
        }
      });
      headroom.init();
      if(window.location.hash)
        headroom.unpin();
      $(header).addClass('headroom--transition');

      // offset scroll location for banner on hash change
      // (see: https://github.com/WickyNilliams/headroom.js/issues/38)
      window.addEventListener("hashchange", function(event) {
        window.scrollTo(0, window.pageYOffset - (headerHeight + 25));
      });

      // responsive menu
      $('.distill-site-header').each(function(i, val) {
        var topnav = $(this);
        var toggle = topnav.find('.nav-toggle');
        toggle.on('click', function() {
          topnav.toggleClass('responsive');
        });
      });

      // nav dropdowns
      $('.nav-dropbtn').click(function(e) {
        $(this).next('.nav-dropdown-content').toggleClass('nav-dropdown-active');
        $(this).parent().siblings('.nav-dropdown')
           .children('.nav-dropdown-content').removeClass('nav-dropdown-active');
      });
      $("body").click(function(e){
        $('.nav-dropdown-content').removeClass('nav-dropdown-active');
      });
      $(".nav-dropdown").click(function(e){
        e.stopPropagation();
      });
    });
  </script>

  <style type="text/css">

  /* Theme (user-documented overrideables for nav appearance) */

  .distill-site-nav {
    color: rgba(255, 255, 255, 0.8);
    background-color: #0F2E3D;
    font-size: 15px;
    font-weight: 300;
  }

  .distill-site-nav a {
    color: inherit;
    text-decoration: none;
  }

  .distill-site-nav a:hover {
    color: white;
  }

  @media print {
    .distill-site-nav {
      display: none;
    }
  }

  .distill-site-header {

  }

  .distill-site-footer {

  }


  /* Site Header */

  .distill-site-header {
    width: 100%;
    box-sizing: border-box;
    z-index: 3;
  }

  .distill-site-header .nav-left {
    display: inline-block;
    margin-left: 8px;
  }

  @media screen and (max-width: 768px) {
    .distill-site-header .nav-left {
      margin-left: 0;
    }
  }


  .distill-site-header .nav-right {
    float: right;
    margin-right: 8px;
  }

  .distill-site-header a,
  .distill-site-header .title {
    display: inline-block;
    text-align: center;
    padding: 14px 10px 14px 10px;
  }

  .distill-site-header .title {
    font-size: 18px;
    min-width: 150px;
  }

  .distill-site-header .logo {
    padding: 0;
  }

  .distill-site-header .logo img {
    display: none;
    max-height: 20px;
    width: auto;
    margin-bottom: -4px;
  }

  .distill-site-header .nav-image img {
    max-height: 18px;
    width: auto;
    display: inline-block;
    margin-bottom: -3px;
  }



  @media screen and (min-width: 1000px) {
    .distill-site-header .logo img {
      display: inline-block;
    }
    .distill-site-header .nav-left {
      margin-left: 20px;
    }
    .distill-site-header .nav-right {
      margin-right: 20px;
    }
    .distill-site-header .title {
      padding-left: 12px;
    }
  }


  .distill-site-header .nav-toggle {
    display: none;
  }

  .nav-dropdown {
    display: inline-block;
    position: relative;
  }

  .nav-dropdown .nav-dropbtn {
    border: none;
    outline: none;
    color: rgba(255, 255, 255, 0.8);
    padding: 16px 10px;
    background-color: transparent;
    font-family: inherit;
    font-size: inherit;
    font-weight: inherit;
    margin: 0;
    margin-top: 1px;
    z-index: 2;
  }

  .nav-dropdown-content {
    display: none;
    position: absolute;
    background-color: white;
    min-width: 200px;
    border: 1px solid rgba(0,0,0,0.15);
    border-radius: 4px;
    box-shadow: 0px 8px 16px 0px rgba(0,0,0,0.1);
    z-index: 1;
    margin-top: 2px;
    white-space: nowrap;
    padding-top: 4px;
    padding-bottom: 4px;
  }

  .nav-dropdown-content hr {
    margin-top: 4px;
    margin-bottom: 4px;
    border: none;
    border-bottom: 1px solid rgba(0, 0, 0, 0.1);
  }

  .nav-dropdown-active {
    display: block;
  }

  .nav-dropdown-content a, .nav-dropdown-content .nav-dropdown-header {
    color: black;
    padding: 6px 24px;
    text-decoration: none;
    display: block;
    text-align: left;
  }

  .nav-dropdown-content .nav-dropdown-header {
    display: block;
    padding: 5px 24px;
    padding-bottom: 0;
    text-transform: uppercase;
    font-size: 14px;
    color: #999999;
    white-space: nowrap;
  }

  .nav-dropdown:hover .nav-dropbtn {
    color: white;
  }

  .nav-dropdown-content a:hover {
    background-color: #ddd;
    color: black;
  }

  .nav-right .nav-dropdown-content {
    margin-left: -45%;
    right: 0;
  }

  @media screen and (max-width: 768px) {
    .distill-site-header a, .distill-site-header .nav-dropdown  {display: none;}
    .distill-site-header a.nav-toggle {
      float: right;
      display: block;
    }
    .distill-site-header .title {
      margin-left: 0;
    }
    .distill-site-header .nav-right {
      margin-right: 0;
    }
    .distill-site-header {
      overflow: hidden;
    }
    .nav-right .nav-dropdown-content {
      margin-left: 0;
    }
  }


  @media screen and (max-width: 768px) {
    .distill-site-header.responsive {position: relative; min-height: 500px; }
    .distill-site-header.responsive a.nav-toggle {
      position: absolute;
      right: 0;
      top: 0;
    }
    .distill-site-header.responsive a,
    .distill-site-header.responsive .nav-dropdown {
      display: block;
      text-align: left;
    }
    .distill-site-header.responsive .nav-left,
    .distill-site-header.responsive .nav-right {
      width: 100%;
    }
    .distill-site-header.responsive .nav-dropdown {float: none;}
    .distill-site-header.responsive .nav-dropdown-content {position: relative;}
    .distill-site-header.responsive .nav-dropdown .nav-dropbtn {
      display: block;
      width: 100%;
      text-align: left;
    }
  }

  /* Site Footer */

  .distill-site-footer {
    width: 100%;
    overflow: hidden;
    box-sizing: border-box;
    z-index: 3;
    margin-top: 30px;
    padding-top: 30px;
    padding-bottom: 30px;
    text-align: center;
  }

  /* Headroom */

  d-title {
    padding-top: 6rem;
  }

  @media print {
    d-title {
      padding-top: 4rem;
    }
  }

  .headroom {
    z-index: 1000;
    position: fixed;
    top: 0;
    left: 0;
    right: 0;
  }

  .headroom--transition {
    transition: all .4s ease-in-out;
  }

  .headroom--unpinned {
    top: -100px;
  }

  .headroom--pinned {
    top: 0;
  }

  /* adjust viewport for navbar height */
  /* helps vertically center bootstrap (non-distill) content */
  .min-vh-100 {
    min-height: calc(100vh - 100px) !important;
  }

  </style>

  <script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
  <link href="site_libs/font-awesome-6.2.1/css/all.min.css" rel="stylesheet"/>
  <link href="site_libs/font-awesome-6.2.1/css/v4-shims.min.css" rel="stylesheet"/>
  <script src="site_libs/headroom-0.9.4/headroom.min.js"></script>
  <!--/radix_placeholder_navigation_in_header-->
  <!--radix_placeholder_distill-->

  <style type="text/css">

  body {
    background-color: white;
  }

  .pandoc-table {
    width: 100%;
  }

  .pandoc-table>caption {
    margin-bottom: 10px;
  }

  .pandoc-table th:not([align]) {
    text-align: left;
  }

  .pagedtable-footer {
    font-size: 15px;
  }

  d-byline .byline {
    grid-template-columns: 2fr 2fr;
  }

  d-byline .byline h3 {
    margin-block-start: 1.5em;
  }

  d-byline .byline .authors-affiliations h3 {
    margin-block-start: 0.5em;
  }

  .authors-affiliations .orcid-id {
    width: 16px;
    height:16px;
    margin-left: 4px;
    margin-right: 4px;
    vertical-align: middle;
    padding-bottom: 2px;
  }

  d-title .dt-tags {
    margin-top: 1em;
    grid-column: text;
  }

  .dt-tags .dt-tag {
    text-decoration: none;
    display: inline-block;
    color: rgba(0,0,0,0.6);
    padding: 0em 0.4em;
    margin-right: 0.5em;
    margin-bottom: 0.4em;
    font-size: 70%;
    border: 1px solid rgba(0,0,0,0.2);
    border-radius: 3px;
    text-transform: uppercase;
    font-weight: 500;
  }

  d-article table.gt_table td,
  d-article table.gt_table th {
    border-bottom: none;
    font-size: 100%;
  }

  .html-widget {
    margin-bottom: 2.0em;
  }

  .l-screen-inset {
    padding-right: 16px;
  }

  .l-screen .caption {
    margin-left: 10px;
  }

  .shaded {
    background: rgb(247, 247, 247);
    padding-top: 20px;
    padding-bottom: 20px;
    border-top: 1px solid rgba(0, 0, 0, 0.1);
    border-bottom: 1px solid rgba(0, 0, 0, 0.1);
  }

  .shaded .html-widget {
    margin-bottom: 0;
    border: 1px solid rgba(0, 0, 0, 0.1);
  }

  .shaded .shaded-content {
    background: white;
  }

  .text-output {
    margin-top: 0;
    line-height: 1.5em;
  }

  .hidden {
    display: none !important;
  }

  d-article {
    padding-top: 2.5rem;
    padding-bottom: 30px;
  }

  d-appendix {
    padding-top: 30px;
  }

  d-article>p>img {
    width: 100%;
  }

  d-article h2 {
    margin: 1rem 0 1.5rem 0;
  }

  d-article h3 {
    margin-top: 1.5rem;
  }

  d-article iframe {
    border: 1px solid rgba(0, 0, 0, 0.1);
    margin-bottom: 2.0em;
    width: 100%;
  }

  /* Tweak code blocks */

  d-article div.sourceCode code,
  d-article pre code {
    font-family: Consolas, Monaco, 'Andale Mono', 'Ubuntu Mono', monospace;
  }

  d-article pre,
  d-article div.sourceCode,
  d-article div.sourceCode pre {
    overflow: auto;
  }

  d-article div.sourceCode {
    background-color: white;
  }

  d-article div.sourceCode pre {
    padding-left: 10px;
    font-size: 12px;
    border-left: 2px solid rgba(0,0,0,0.1);
  }

  d-article pre {
    font-size: 12px;
    color: black;
    background: none;
    margin-top: 0;
    text-align: left;
    white-space: pre;
    word-spacing: normal;
    word-break: normal;
    word-wrap: normal;
    line-height: 1.5;

    -moz-tab-size: 4;
    -o-tab-size: 4;
    tab-size: 4;

    -webkit-hyphens: none;
    -moz-hyphens: none;
    -ms-hyphens: none;
    hyphens: none;
  }

  d-article pre a {
    border-bottom: none;
  }

  d-article pre a:hover {
    border-bottom: none;
    text-decoration: underline;
  }

  d-article details {
    grid-column: text;
    margin-bottom: 0.8em;
  }

  @media(min-width: 768px) {

  d-article pre,
  d-article div.sourceCode,
  d-article div.sourceCode pre {
    overflow: visible !important;
  }

  d-article div.sourceCode pre {
    padding-left: 18px;
    font-size: 14px;
  }

  d-article pre {
    font-size: 14px;
  }

  }

  figure img.external {
    background: white;
    border: 1px solid rgba(0, 0, 0, 0.1);
    box-shadow: 0 1px 8px rgba(0, 0, 0, 0.1);
    padding: 18px;
    box-sizing: border-box;
  }

  /* CSS for d-contents */

  .d-contents {
    grid-column: text;
    color: rgba(0,0,0,0.8);
    font-size: 0.9em;
    padding-bottom: 1em;
    margin-bottom: 1em;
    padding-bottom: 0.5em;
    margin-bottom: 1em;
    padding-left: 0.25em;
    justify-self: start;
  }

  @media(min-width: 1000px) {
    .d-contents.d-contents-float {
      height: 0;
      grid-column-start: 1;
      grid-column-end: 4;
      justify-self: center;
      padding-right: 3em;
      padding-left: 2em;
    }
  }

  .d-contents nav h3 {
    font-size: 18px;
    margin-top: 0;
    margin-bottom: 1em;
  }

  .d-contents li {
    list-style-type: none
  }

  .d-contents nav > ul {
    padding-left: 0;
  }

  .d-contents ul {
    padding-left: 1em
  }

  .d-contents nav ul li {
    margin-top: 0.6em;
    margin-bottom: 0.2em;
  }

  .d-contents nav a {
    font-size: 13px;
    border-bottom: none;
    text-decoration: none
    color: rgba(0, 0, 0, 0.8);
  }

  .d-contents nav a:hover {
    text-decoration: underline solid rgba(0, 0, 0, 0.6)
  }

  .d-contents nav > ul > li > a {
    font-weight: 600;
  }

  .d-contents nav > ul > li > ul {
    font-weight: inherit;
  }

  .d-contents nav > ul > li > ul > li {
    margin-top: 0.2em;
  }


  .d-contents nav ul {
    margin-top: 0;
    margin-bottom: 0.25em;
  }

  .d-article-with-toc h2:nth-child(2) {
    margin-top: 0;
  }


  /* Figure */

  .figure {
    position: relative;
    margin-bottom: 2.5em;
    margin-top: 1.5em;
  }

  .figure .caption {
    color: rgba(0, 0, 0, 0.6);
    font-size: 12px;
    line-height: 1.5em;
  }

  .figure img.external {
    background: white;
    border: 1px solid rgba(0, 0, 0, 0.1);
    box-shadow: 0 1px 8px rgba(0, 0, 0, 0.1);
    padding: 18px;
    box-sizing: border-box;
  }

  .figure .caption a {
    color: rgba(0, 0, 0, 0.6);
  }

  .figure .caption b,
  .figure .caption strong, {
    font-weight: 600;
    color: rgba(0, 0, 0, 1.0);
  }

  /* Citations */

  d-article .citation {
    color: inherit;
    cursor: inherit;
  }

  div.hanging-indent{
    margin-left: 1em; text-indent: -1em;
  }

  /* Citation hover box */

  .tippy-box[data-theme~=light-border] {
    background-color: rgba(250, 250, 250, 0.95);
  }

  .tippy-content > p {
    margin-bottom: 0;
    padding: 2px;
  }


  /* Tweak 1000px media break to show more text */

  @media(min-width: 1000px) {
    .base-grid,
    distill-header,
    d-title,
    d-abstract,
    d-article,
    d-appendix,
    distill-appendix,
    d-byline,
    d-footnote-list,
    d-citation-list,
    distill-footer {
      grid-template-columns: [screen-start] 1fr [page-start kicker-start] 80px [middle-start] 50px [text-start kicker-end] 65px 65px 65px 65px 65px 65px 65px 65px [text-end gutter-start] 65px [middle-end] 65px [page-end gutter-end] 1fr [screen-end];
      grid-column-gap: 16px;
    }

    .grid {
      grid-column-gap: 16px;
    }

    d-article {
      font-size: 1.06rem;
      line-height: 1.7em;
    }
    figure .caption, .figure .caption, figure figcaption {
      font-size: 13px;
    }
  }

  @media(min-width: 1180px) {
    .base-grid,
    distill-header,
    d-title,
    d-abstract,
    d-article,
    d-appendix,
    distill-appendix,
    d-byline,
    d-footnote-list,
    d-citation-list,
    distill-footer {
      grid-template-columns: [screen-start] 1fr [page-start kicker-start] 60px [middle-start] 60px [text-start kicker-end] 60px 60px 60px 60px 60px 60px 60px 60px [text-end gutter-start] 60px [middle-end] 60px [page-end gutter-end] 1fr [screen-end];
      grid-column-gap: 32px;
    }

    .grid {
      grid-column-gap: 32px;
    }
  }


  /* Get the citation styles for the appendix (not auto-injected on render since
     we do our own rendering of the citation appendix) */

  d-appendix .citation-appendix,
  .d-appendix .citation-appendix {
    font-size: 11px;
    line-height: 15px;
    border-left: 1px solid rgba(0, 0, 0, 0.1);
    padding-left: 18px;
    border: 1px solid rgba(0,0,0,0.1);
    background: rgba(0, 0, 0, 0.02);
    padding: 10px 18px;
    border-radius: 3px;
    color: rgba(150, 150, 150, 1);
    overflow: hidden;
    margin-top: -12px;
    white-space: pre-wrap;
    word-wrap: break-word;
  }

  /* Include appendix styles here so they can be overridden */

  d-appendix {
    contain: layout style;
    font-size: 0.8em;
    line-height: 1.7em;
    margin-top: 60px;
    margin-bottom: 0;
    border-top: 1px solid rgba(0, 0, 0, 0.1);
    color: rgba(0,0,0,0.5);
    padding-top: 60px;
    padding-bottom: 48px;
  }

  d-appendix h3 {
    grid-column: page-start / text-start;
    font-size: 15px;
    font-weight: 500;
    margin-top: 1em;
    margin-bottom: 0;
    color: rgba(0,0,0,0.65);
  }

  d-appendix h3 + * {
    margin-top: 1em;
  }

  d-appendix ol {
    padding: 0 0 0 15px;
  }

  @media (min-width: 768px) {
    d-appendix ol {
      padding: 0 0 0 30px;
      margin-left: -30px;
    }
  }

  d-appendix li {
    margin-bottom: 1em;
  }

  d-appendix a {
    color: rgba(0, 0, 0, 0.6);
  }

  d-appendix > * {
    grid-column: text;
  }

  d-appendix > d-footnote-list,
  d-appendix > d-citation-list,
  d-appendix > distill-appendix {
    grid-column: screen;
  }

  /* Include footnote styles here so they can be overridden */

  d-footnote-list {
    contain: layout style;
  }

  d-footnote-list > * {
    grid-column: text;
  }

  d-footnote-list a.footnote-backlink {
    color: rgba(0,0,0,0.3);
    padding-left: 0.5em;
  }



  /* Anchor.js */

  .anchorjs-link {
    /*transition: all .25s linear; */
    text-decoration: none;
    border-bottom: none;
  }
  *:hover > .anchorjs-link {
    margin-left: -1.125em !important;
    text-decoration: none;
    border-bottom: none;
  }

  /* Social footer */

  .social_footer {
    margin-top: 30px;
    margin-bottom: 0;
    color: rgba(0,0,0,0.67);
  }

  .disqus-comments {
    margin-right: 30px;
  }

  .disqus-comment-count {
    border-bottom: 1px solid rgba(0, 0, 0, 0.4);
    cursor: pointer;
  }

  #disqus_thread {
    margin-top: 30px;
  }

  .article-sharing a {
    border-bottom: none;
    margin-right: 8px;
  }

  .article-sharing a:hover {
    border-bottom: none;
  }

  .sidebar-section.subscribe {
    font-size: 12px;
    line-height: 1.6em;
  }

  .subscribe p {
    margin-bottom: 0.5em;
  }


  .article-footer .subscribe {
    font-size: 15px;
    margin-top: 45px;
  }


  .sidebar-section.custom {
    font-size: 12px;
    line-height: 1.6em;
  }

  .custom p {
    margin-bottom: 0.5em;
  }

  /* Styles for listing layout (hide title) */
  .layout-listing d-title, .layout-listing .d-title {
    display: none;
  }

  /* Styles for posts lists (not auto-injected) */


  .posts-with-sidebar {
    padding-left: 45px;
    padding-right: 45px;
  }

  .posts-list .description h2,
  .posts-list .description p {
    font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen, Ubuntu, Cantarell, "Fira Sans", "Droid Sans", "Helvetica Neue", Arial, sans-serif;
  }

  .posts-list .description h2 {
    font-weight: 700;
    border-bottom: none;
    padding-bottom: 0;
  }

  .posts-list h2.post-tag {
    border-bottom: 1px solid rgba(0, 0, 0, 0.2);
    padding-bottom: 12px;
  }
  .posts-list {
    margin-top: 60px;
    margin-bottom: 24px;
  }

  .posts-list .post-preview {
    text-decoration: none;
    overflow: hidden;
    display: block;
    border-bottom: 1px solid rgba(0, 0, 0, 0.1);
    padding: 24px 0;
  }

  .post-preview-last {
    border-bottom: none !important;
  }

  .posts-list .posts-list-caption {
    grid-column: screen;
    font-weight: 400;
  }

  .posts-list .post-preview h2 {
    margin: 0 0 6px 0;
    line-height: 1.2em;
    font-style: normal;
    font-size: 24px;
  }

  .posts-list .post-preview p {
    margin: 0 0 12px 0;
    line-height: 1.4em;
    font-size: 16px;
  }

  .posts-list .post-preview .thumbnail {
    box-sizing: border-box;
    margin-bottom: 24px;
    position: relative;
    max-width: 500px;
  }
  .posts-list .post-preview img {
    width: 100%;
    display: block;
  }

  .posts-list .metadata {
    font-size: 12px;
    line-height: 1.4em;
    margin-bottom: 18px;
  }

  .posts-list .metadata > * {
    display: inline-block;
  }

  .posts-list .metadata .publishedDate {
    margin-right: 2em;
  }

  .posts-list .metadata .dt-authors {
    display: block;
    margin-top: 0.3em;
    margin-right: 2em;
  }

  .posts-list .dt-tags {
    display: block;
    line-height: 1em;
  }

  .posts-list .dt-tags .dt-tag {
    display: inline-block;
    color: rgba(0,0,0,0.6);
    padding: 0.3em 0.4em;
    margin-right: 0.2em;
    margin-bottom: 0.4em;
    font-size: 60%;
    border: 1px solid rgba(0,0,0,0.2);
    border-radius: 3px;
    text-transform: uppercase;
    font-weight: 500;
  }

  .posts-list img {
    opacity: 1;
  }

  .posts-list img[data-src] {
    opacity: 0;
  }

  .posts-more {
    clear: both;
  }


  .posts-sidebar {
    font-size: 16px;
  }

  .posts-sidebar h3 {
    font-size: 16px;
    margin-top: 0;
    margin-bottom: 0.5em;
    font-weight: 400;
    text-transform: uppercase;
  }

  .sidebar-section {
    margin-bottom: 30px;
  }

  .categories ul {
    list-style-type: none;
    margin: 0;
    padding: 0;
  }

  .categories li {
    color: rgba(0, 0, 0, 0.8);
    margin-bottom: 0;
  }

  .categories li>a {
    border-bottom: none;
  }

  .categories li>a:hover {
    border-bottom: 1px solid rgba(0, 0, 0, 0.4);
  }

  .categories .active {
    font-weight: 600;
  }

  .categories .category-count {
    color: rgba(0, 0, 0, 0.4);
  }


  @media(min-width: 768px) {
    .posts-list .post-preview h2 {
      font-size: 26px;
    }
    .posts-list .post-preview .thumbnail {
      float: right;
      width: 30%;
      margin-bottom: 0;
    }
    .posts-list .post-preview .description {
      float: left;
      width: 45%;
    }
    .posts-list .post-preview .metadata {
      float: left;
      width: 20%;
      margin-top: 8px;
    }
    .posts-list .post-preview p {
      margin: 0 0 12px 0;
      line-height: 1.5em;
      font-size: 16px;
    }
    .posts-with-sidebar .posts-list {
      float: left;
      width: 75%;
    }
    .posts-with-sidebar .posts-sidebar {
      float: right;
      width: 20%;
      margin-top: 60px;
      padding-top: 24px;
      padding-bottom: 24px;
    }
  }


  /* Improve display for browsers without grid (IE/Edge <= 15) */

  .downlevel {
    line-height: 1.6em;
    font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen, Ubuntu, Cantarell, "Fira Sans", "Droid Sans", "Helvetica Neue", Arial, sans-serif;
    margin: 0;
  }

  .downlevel .d-title {
    padding-top: 6rem;
    padding-bottom: 1.5rem;
  }

  .downlevel .d-title h1 {
    font-size: 50px;
    font-weight: 700;
    line-height: 1.1em;
    margin: 0 0 0.5rem;
  }

  .downlevel .d-title p {
    font-weight: 300;
    font-size: 1.2rem;
    line-height: 1.55em;
    margin-top: 0;
  }

  .downlevel .d-byline {
    padding-top: 0.8em;
    padding-bottom: 0.8em;
    font-size: 0.8rem;
    line-height: 1.8em;
  }

  .downlevel .section-separator {
    border: none;
    border-top: 1px solid rgba(0, 0, 0, 0.1);
  }

  .downlevel .d-article {
    font-size: 1.06rem;
    line-height: 1.7em;
    padding-top: 1rem;
    padding-bottom: 2rem;
  }


  .downlevel .d-appendix {
    padding-left: 0;
    padding-right: 0;
    max-width: none;
    font-size: 0.8em;
    line-height: 1.7em;
    margin-bottom: 0;
    color: rgba(0,0,0,0.5);
    padding-top: 40px;
    padding-bottom: 48px;
  }

  .downlevel .footnotes ol {
    padding-left: 13px;
  }

  .downlevel .base-grid,
  .downlevel .distill-header,
  .downlevel .d-title,
  .downlevel .d-abstract,
  .downlevel .d-article,
  .downlevel .d-appendix,
  .downlevel .distill-appendix,
  .downlevel .d-byline,
  .downlevel .d-footnote-list,
  .downlevel .d-citation-list,
  .downlevel .distill-footer,
  .downlevel .appendix-bottom,
  .downlevel .posts-container {
    padding-left: 40px;
    padding-right: 40px;
  }

  @media(min-width: 768px) {
    .downlevel .base-grid,
    .downlevel .distill-header,
    .downlevel .d-title,
    .downlevel .d-abstract,
    .downlevel .d-article,
    .downlevel .d-appendix,
    .downlevel .distill-appendix,
    .downlevel .d-byline,
    .downlevel .d-footnote-list,
    .downlevel .d-citation-list,
    .downlevel .distill-footer,
    .downlevel .appendix-bottom,
    .downlevel .posts-container {
    padding-left: 150px;
    padding-right: 150px;
    max-width: 900px;
  }
  }

  .downlevel pre code {
    display: block;
    border-left: 2px solid rgba(0, 0, 0, .1);
    padding: 0 0 0 20px;
    font-size: 14px;
  }

  .downlevel code, .downlevel pre {
    color: black;
    background: none;
    font-family: Consolas, Monaco, 'Andale Mono', 'Ubuntu Mono', monospace;
    text-align: left;
    white-space: pre;
    word-spacing: normal;
    word-break: normal;
    word-wrap: normal;
    line-height: 1.5;

    -moz-tab-size: 4;
    -o-tab-size: 4;
    tab-size: 4;

    -webkit-hyphens: none;
    -moz-hyphens: none;
    -ms-hyphens: none;
    hyphens: none;
  }

  .downlevel .posts-list .post-preview {
    color: inherit;
  }



  </style>

  <script type="application/javascript">

  function is_downlevel_browser() {
    if (bowser.isUnsupportedBrowser({ msie: "12", msedge: "16"},
                                   window.navigator.userAgent)) {
      return true;
    } else {
      return window.load_distill_framework === undefined;
    }
  }

  // show body when load is complete
  function on_load_complete() {

    // add anchors
    if (window.anchors) {
      window.anchors.options.placement = 'left';
      window.anchors.add('d-article > h2, d-article > h3, d-article > h4, d-article > h5');
    }


    // set body to visible
    document.body.style.visibility = 'visible';

    // force redraw for leaflet widgets
    if (window.HTMLWidgets) {
      var maps = window.HTMLWidgets.findAll(".leaflet");
      $.each(maps, function(i, el) {
        var map = this.getMap();
        map.invalidateSize();
        map.eachLayer(function(layer) {
          if (layer instanceof L.TileLayer)
            layer.redraw();
        });
      });
    }

    // trigger 'shown' so htmlwidgets resize
    $('d-article').trigger('shown');
  }

  function init_distill() {

    init_common();

    // create front matter
    var front_matter = $('<d-front-matter></d-front-matter>');
    $('#distill-front-matter').wrap(front_matter);

    // create d-title
    $('.d-title').changeElementType('d-title');

    // create d-byline
    var byline = $('<d-byline></d-byline>');
    $('.d-byline').replaceWith(byline);

    // create d-article
    var article = $('<d-article></d-article>');
    $('.d-article').wrap(article).children().unwrap();

    // move posts container into article
    $('.posts-container').appendTo($('d-article'));

    // create d-appendix
    $('.d-appendix').changeElementType('d-appendix');

    // flag indicating that we have appendix items
    var appendix = $('.appendix-bottom').children('h3').length > 0;

    // replace footnotes with <d-footnote>
    $('.footnote-ref').each(function(i, val) {
      appendix = true;
      var href = $(this).attr('href');
      var id = href.replace('#', '');
      var fn = $('#' + id);
      var fn_p = $('#' + id + '>p');
      fn_p.find('.footnote-back').remove();
      var text = fn_p.html();
      var dtfn = $('<d-footnote></d-footnote>');
      dtfn.html(text);
      $(this).replaceWith(dtfn);
    });
    // remove footnotes
    $('.footnotes').remove();

    // move refs into #references-listing
    $('#references-listing').replaceWith($('#refs'));

    $('h1.appendix, h2.appendix').each(function(i, val) {
      $(this).changeElementType('h3');
    });
    $('h3.appendix').each(function(i, val) {
      var id = $(this).attr('id');
      $('.d-contents a[href="#' + id + '"]').parent().remove();
      appendix = true;
      $(this).nextUntil($('h1, h2, h3')).addBack().appendTo($('d-appendix'));
    });

    // show d-appendix if we have appendix content
    $("d-appendix").css('display', appendix ? 'grid' : 'none');

    // localize layout chunks to just output
    $('.layout-chunk').each(function(i, val) {

      // capture layout
      var layout = $(this).attr('data-layout');

      // apply layout to markdown level block elements
      var elements = $(this).children().not('details, div.sourceCode, pre, script');
      elements.each(function(i, el) {
        var layout_div = $('<div class="' + layout + '"></div>');
        if (layout_div.hasClass('shaded')) {
          var shaded_content = $('<div class="shaded-content"></div>');
          $(this).wrap(shaded_content);
          $(this).parent().wrap(layout_div);
        } else {
          $(this).wrap(layout_div);
        }
      });


      // unwrap the layout-chunk div
      $(this).children().unwrap();
    });

    // remove code block used to force  highlighting css
    $('.distill-force-highlighting-css').parent().remove();

    // remove empty line numbers inserted by pandoc when using a
    // custom syntax highlighting theme
    $('code.sourceCode a:empty').remove();

    // load distill framework
    load_distill_framework();

    // wait for window.distillRunlevel == 4 to do post processing
    function distill_post_process() {

      if (!window.distillRunlevel || window.distillRunlevel < 4)
        return;

      // hide author/affiliations entirely if we have no authors
      var front_matter = JSON.parse($("#distill-front-matter").html());
      var have_authors = front_matter.authors && front_matter.authors.length > 0;
      if (!have_authors)
        $('d-byline').addClass('hidden');

      // article with toc class
      $('.d-contents').parent().addClass('d-article-with-toc');

      // strip links that point to #
      $('.authors-affiliations').find('a[href="#"]').removeAttr('href');

      // add orcid ids
      $('.authors-affiliations').find('.author').each(function(i, el) {
        var orcid_id = front_matter.authors[i].orcidID;
        if (orcid_id) {
          var a = $('<a></a>');
          a.attr('href', 'https://orcid.org/' + orcid_id);
          var img = $('<img></img>');
          img.addClass('orcid-id');
          img.attr('alt', 'ORCID ID');
          img.attr('src','data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAYAAAAf8/9hAAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAA2ZpVFh0WE1MOmNvbS5hZG9iZS54bXAAAAAAADw/eHBhY2tldCBiZWdpbj0i77u/IiBpZD0iVzVNME1wQ2VoaUh6cmVTek5UY3prYzlkIj8+IDx4OnhtcG1ldGEgeG1sbnM6eD0iYWRvYmU6bnM6bWV0YS8iIHg6eG1wdGs9IkFkb2JlIFhNUCBDb3JlIDUuMC1jMDYwIDYxLjEzNDc3NywgMjAxMC8wMi8xMi0xNzozMjowMCAgICAgICAgIj4gPHJkZjpSREYgeG1sbnM6cmRmPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5LzAyLzIyLXJkZi1zeW50YXgtbnMjIj4gPHJkZjpEZXNjcmlwdGlvbiByZGY6YWJvdXQ9IiIgeG1sbnM6eG1wTU09Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC9tbS8iIHhtbG5zOnN0UmVmPSJodHRwOi8vbnMuYWRvYmUuY29tL3hhcC8xLjAvc1R5cGUvUmVzb3VyY2VSZWYjIiB4bWxuczp4bXA9Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC8iIHhtcE1NOk9yaWdpbmFsRG9jdW1lbnRJRD0ieG1wLmRpZDo1N0NEMjA4MDI1MjA2ODExOTk0QzkzNTEzRjZEQTg1NyIgeG1wTU06RG9jdW1lbnRJRD0ieG1wLmRpZDozM0NDOEJGNEZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wTU06SW5zdGFuY2VJRD0ieG1wLmlpZDozM0NDOEJGM0ZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wOkNyZWF0b3JUb29sPSJBZG9iZSBQaG90b3Nob3AgQ1M1IE1hY2ludG9zaCI+IDx4bXBNTTpEZXJpdmVkRnJvbSBzdFJlZjppbnN0YW5jZUlEPSJ4bXAuaWlkOkZDN0YxMTc0MDcyMDY4MTE5NUZFRDc5MUM2MUUwNEREIiBzdFJlZjpkb2N1bWVudElEPSJ4bXAuZGlkOjU3Q0QyMDgwMjUyMDY4MTE5OTRDOTM1MTNGNkRBODU3Ii8+IDwvcmRmOkRlc2NyaXB0aW9uPiA8L3JkZjpSREY+IDwveDp4bXBtZXRhPiA8P3hwYWNrZXQgZW5kPSJyIj8+84NovQAAAR1JREFUeNpiZEADy85ZJgCpeCB2QJM6AMQLo4yOL0AWZETSqACk1gOxAQN+cAGIA4EGPQBxmJA0nwdpjjQ8xqArmczw5tMHXAaALDgP1QMxAGqzAAPxQACqh4ER6uf5MBlkm0X4EGayMfMw/Pr7Bd2gRBZogMFBrv01hisv5jLsv9nLAPIOMnjy8RDDyYctyAbFM2EJbRQw+aAWw/LzVgx7b+cwCHKqMhjJFCBLOzAR6+lXX84xnHjYyqAo5IUizkRCwIENQQckGSDGY4TVgAPEaraQr2a4/24bSuoExcJCfAEJihXkWDj3ZAKy9EJGaEo8T0QSxkjSwORsCAuDQCD+QILmD1A9kECEZgxDaEZhICIzGcIyEyOl2RkgwAAhkmC+eAm0TAAAAABJRU5ErkJggg==');
          a.append(img);
          $(this).append(a);
        }
      });

      // hide elements of author/affiliations grid that have no value
      function hide_byline_column(caption) {
        $('d-byline').find('h3:contains("' + caption + '")').parent().css('visibility', 'hidden');
      }

      // affiliations
      var have_affiliations = false;
      for (var i = 0; i<front_matter.authors.length; ++i) {
        var author = front_matter.authors[i];
        if (author.affiliation !== "&nbsp;") {
          have_affiliations = true;
          break;
        }
      }
      if (!have_affiliations)
        $('d-byline').find('h3:contains("Affiliations")').css('visibility', 'hidden');

      // published date
      if (!front_matter.publishedDate)
        hide_byline_column("Published");

      // document object identifier
      var doi = $('d-byline').find('h3:contains("DOI")');
      var doi_p = doi.next().empty();
      if (!front_matter.doi) {
        // if we have a citation and valid citationText then link to that
        if ($('#citation').length > 0 && front_matter.citationText) {
          doi.html('Citation');
          $('<a href="#citation"></a>')
            .text(front_matter.citationText)
            .appendTo(doi_p);
        } else {
          hide_byline_column("DOI");
        }
      } else {
        $('<a></a>')
           .attr('href', "https://doi.org/" + front_matter.doi)
           .html(front_matter.doi)
           .appendTo(doi_p);
      }

       // change plural form of authors/affiliations
      if (front_matter.authors.length === 1) {
        var grid = $('.authors-affiliations');
        grid.children('h3:contains("Authors")').text('Author');
        grid.children('h3:contains("Affiliations")').text('Affiliation');
      }

      // remove d-appendix and d-footnote-list local styles
      $('d-appendix > style:first-child').remove();
      $('d-footnote-list > style:first-child').remove();

      // move appendix-bottom entries to the bottom
      $('.appendix-bottom').appendTo('d-appendix').children().unwrap();
      $('.appendix-bottom').remove();

      // hoverable references
      $('span.citation[data-cites]').each(function() {
        const citeChild = $(this).children()[0]
        // Do not process if @xyz has been used without escaping and without bibliography activated
        // https://github.com/rstudio/distill/issues/466
        if (citeChild === undefined) return true

        if (citeChild.nodeName == "D-FOOTNOTE") {
          var fn = citeChild
          $(this).html(fn.shadowRoot.querySelector("sup"))
          $(this).id = fn.id
          fn.remove()
        }
        var refs = $(this).attr('data-cites').split(" ");
        var refHtml = refs.map(function(ref) {
          // Could use CSS.escape too here, we insure backward compatibility in navigator
          return "<p>" + $('div[id="ref-' + ref + '"]').html() + "</p>";
        }).join("\n");
        window.tippy(this, {
          allowHTML: true,
          content: refHtml,
          maxWidth: 500,
          interactive: true,
          interactiveBorder: 10,
          theme: 'light-border',
          placement: 'bottom-start'
        });
      });

      // fix footnotes in tables (#411)
      // replacing broken distill.pub feature
      $('table d-footnote').each(function() {
        // we replace internal showAtNode methode which is triggered when hovering a footnote
        this.hoverBox.showAtNode = function(node) {
          // ported from https://github.com/distillpub/template/pull/105/files
          calcOffset = function(elem) {
              let x = elem.offsetLeft;
              let y = elem.offsetTop;
              // Traverse upwards until an `absolute` element is found or `elem`
              // becomes null.
              while (elem = elem.offsetParent && elem.style.position != 'absolute') {
                  x += elem.offsetLeft;
                  y += elem.offsetTop;
              }

              return { left: x, top: y };
          }
          // https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/offsetTop
          const bbox = node.getBoundingClientRect();
          const offset = calcOffset(node);
          this.show([offset.left + bbox.width, offset.top + bbox.height]);
        }
      })

      // clear polling timer
      clearInterval(tid);

      // show body now that everything is ready
      on_load_complete();
    }

    var tid = setInterval(distill_post_process, 50);
    distill_post_process();

  }

  function init_downlevel() {

    init_common();

     // insert hr after d-title
    $('.d-title').after($('<hr class="section-separator"/>'));

    // check if we have authors
    var front_matter = JSON.parse($("#distill-front-matter").html());
    var have_authors = front_matter.authors && front_matter.authors.length > 0;

    // manage byline/border
    if (!have_authors)
      $('.d-byline').remove();
    $('.d-byline').after($('<hr class="section-separator"/>'));
    $('.d-byline a').remove();

    // remove toc
    $('.d-contents').remove();

    // move appendix elements
    $('h1.appendix, h2.appendix').each(function(i, val) {
      $(this).changeElementType('h3');
    });
    $('h3.appendix').each(function(i, val) {
      $(this).nextUntil($('h1, h2, h3')).addBack().appendTo($('.d-appendix'));
    });


    // inject headers into references and footnotes
    var refs_header = $('<h3></h3>');
    refs_header.text('References');
    $('#refs').prepend(refs_header);

    var footnotes_header = $('<h3></h3');
    footnotes_header.text('Footnotes');
    $('.footnotes').children('hr').first().replaceWith(footnotes_header);

    // move appendix-bottom entries to the bottom
    $('.appendix-bottom').appendTo('.d-appendix').children().unwrap();
    $('.appendix-bottom').remove();

    // remove appendix if it's empty
    if ($('.d-appendix').children().length === 0)
      $('.d-appendix').remove();

    // prepend separator above appendix
    $('.d-appendix').before($('<hr class="section-separator" style="clear: both"/>'));

    // trim code
    $('pre>code').each(function(i, val) {
      $(this).html($.trim($(this).html()));
    });

    // move posts-container right before article
    $('.posts-container').insertBefore($('.d-article'));

    $('body').addClass('downlevel');

    on_load_complete();
  }


  function init_common() {

    // jquery plugin to change element types
    (function($) {
      $.fn.changeElementType = function(newType) {
        var attrs = {};

        $.each(this[0].attributes, function(idx, attr) {
          attrs[attr.nodeName] = attr.nodeValue;
        });

        this.replaceWith(function() {
          return $("<" + newType + "/>", attrs).append($(this).contents());
        });
      };
    })(jQuery);

    // prevent underline for linked images
    $('a > img').parent().css({'border-bottom' : 'none'});

    // mark non-body figures created by knitr chunks as 100% width
    $('.layout-chunk').each(function(i, val) {
      var figures = $(this).find('img, .html-widget');
      // ignore leaflet img layers (#106)
      figures = figures.filter(':not(img[class*="leaflet"])')
      if ($(this).attr('data-layout') !== "l-body") {
        figures.css('width', '100%');
      } else {
        figures.css('max-width', '100%');
        figures.filter("[width]").each(function(i, val) {
          var fig = $(this);
          fig.css('width', fig.attr('width') + 'px');
        });

      }
    });

    // auto-append index.html to post-preview links in file: protocol
    // and in rstudio ide preview
    $('.post-preview').each(function(i, val) {
      if (window.location.protocol === "file:")
        $(this).attr('href', $(this).attr('href') + "index.html");
    });

    // get rid of index.html references in header
    if (window.location.protocol !== "file:") {
      $('.distill-site-header a[href]').each(function(i,val) {
        $(this).attr('href', $(this).attr('href').replace(/^index[.]html/, "./"));
      });
    }

    // add class to pandoc style tables
    $('tr.header').parent('thead').parent('table').addClass('pandoc-table');
    $('.kable-table').children('table').addClass('pandoc-table');

    // add figcaption style to table captions
    $('caption').parent('table').addClass("figcaption");

    // initialize posts list
    if (window.init_posts_list)
      window.init_posts_list();

    // implmement disqus comment link
    $('.disqus-comment-count').click(function() {
      window.headroom_prevent_pin = true;
      $('#disqus_thread').toggleClass('hidden');
      if (!$('#disqus_thread').hasClass('hidden')) {
        var offset = $(this).offset();
        $(window).resize();
        $('html, body').animate({
          scrollTop: offset.top - 35
        });
      }
    });
  }

  document.addEventListener('DOMContentLoaded', function() {
    if (is_downlevel_browser())
      init_downlevel();
    else
      window.addEventListener('WebComponentsReady', init_distill);
  });

  </script>

  <!--/radix_placeholder_distill-->
  <script src="site_libs/header-attrs-2.21/header-attrs.js"></script>
  <script src="site_libs/popper-2.6.0/popper.min.js"></script>
  <link href="site_libs/tippy-6.2.7/tippy.css" rel="stylesheet" />
  <link href="site_libs/tippy-6.2.7/tippy-light-border.css" rel="stylesheet" />
  <script src="site_libs/tippy-6.2.7/tippy.umd.min.js"></script>
  <script src="site_libs/anchor-4.2.2/anchor.min.js"></script>
  <script src="site_libs/bowser-1.9.3/bowser.min.js"></script>
  <script src="site_libs/webcomponents-2.0.0/webcomponents.js"></script>
  <script src="site_libs/distill-2.2.21/template.v2.js"></script>
  <!--radix_placeholder_site_in_header-->
  <style type="text/css">
  @import url('https://fonts.googleapis.com/css2?family=Cormorant+Garamond&display=swap');

  @import url('https://fonts.googleapis.com/css2?family=Raleway&display=swap');

  @import url('https://fonts.googleapis.com/css2?family=Source+Code+Pro&display=swap');

  @import url('https://fonts.googleapis.com/css2?family=Bebas+Neue&display=swap');



  .nav-dropdown-content {
    color: #F0A202;
    background-color: #14213d;
  }

  .nav-dropdown-content a:hover {
    background-color: #14213d;
  }

  .nav-dropdown-content a{
    color: #F0A202;
  }

  .nav-dropdown .nav-dropbtn {
    color: #F0A202;
  }

  .distill-site-nav {
    color: #F0A202;
    font-weight: bold;
    background-color: #14213d;
  }

  .distill-site-nav a:hover {
    font-weight: bold;
    color: #FFFFFF;
  }

  body {
    background-color: #FFFFFF;
  }

  h1 {
    font-family: 'Bebas Neue', cursive;
    color: #283044;
  }

  h2 {
    color: #283044;
  }

  d-article h2 {
  	border-bottom: 0;
    padding-bottom: 0;
  }

  /* Use specific font in the body of the text */
  html, body, p {
    
    font-family: 'Raleway', sans-serif;

    
  }
  </style>
  <!--/radix_placeholder_site_in_header-->

  <link rel="stylesheet" href="styles.css" type="text/css"/>

</head>

<body>

<!--radix_placeholder_front_matter-->

<script id="distill-front-matter" type="text/json">
{"title":"Toronto Workshop on Reproducibility","description":"An annual workshop focused on reproducibility in data science and statistics. First held 25-26 February 2021, and again 23-25 February 2022. Free and hosted via Zoom. Jointly hosted by CANSSI Ontario and the Data Sciences Institute. Supported by the Faculty of Information and the Department of Statistical Sciences.","authors":[]}
</script>

<!--/radix_placeholder_front_matter-->
<!--radix_placeholder_navigation_before_body-->
<header class="header header--fixed" role="banner">
<nav class="distill-site-nav distill-site-header">
<div class="nav-left">
<a href="index.html" class="title">Rohan Alexander</a>
</div>
<div class="nav-right">
<a href="index.html">Home</a>
<a href="teaching.html">Teaching</a>
<a href="toronto_data_workshop.html">Toronto Data Workshop</a>
<a href="blog.html">Blog</a>
<div class="nav-dropdown">
<button class="nav-dropbtn">
Other
 
<span class="down-arrow">&#x25BE;</span>
</button>
<div class="nav-dropdown-content">
<a href="bookshelf.html">Bookshelf</a>
<a href="reproducibility.html">Toronto Workshop on Reproducibility</a>
</div>
</div>
<a href="javascript:void(0);" class="nav-toggle">&#9776;</a>
</div>
</nav>
</header>
<!--/radix_placeholder_navigation_before_body-->
<!--radix_placeholder_site_before_body-->
<!--/radix_placeholder_site_before_body-->

<div class="d-title">
<h1>Toronto Workshop on Reproducibility</h1>
<!--radix_placeholder_categories-->
<!--/radix_placeholder_categories-->
<p><p>An annual workshop focused on reproducibility in data science and statistics. First held 25-26 February 2021, and again 23-25 February 2022. Free and hosted via Zoom. Jointly hosted by CANSSI Ontario and the Data Sciences Institute. Supported by the Faculty of Information and the Department of Statistical Sciences.</p></p>
</div>


<div class="d-article">
<div class="d-contents d-contents-float">
<nav class="l-text toc figcaption" id="TOC">
<h3>Contents</h3>
<ul>
<li><a href="#overview" id="toc-overview">Overview</a></li>
<li><a href="#section" id="toc-section">2022</a></li>
<li><a href="#section-1" id="toc-section-1">2021</a></li>
<li><a href="#code-of-conduct" id="toc-code-of-conduct">Code of conduct</a></li>
</ul>
</nav>
</div>
<div class="layout-chunk" data-layout="l-body">
<p><img src="images/information_vert.png" width="25%" /><img src="images/doss_vert.png" width="25%" /><img src="images/canssi_logo_Ontario-RGB-M.png" width="25%" /></p>
</div>
<h2 id="overview">Overview</h2>
<p>This conference brings together academic and industry participants on the critical issue of reproducibility in applied statistics and related areas. The conference is free and hosted online. Everyone is welcome, you don’t need to be affiliated with a university.</p>
<p>The conference has three broad areas of focus:</p>
<ul>
<li><strong>Evaluating reproducibility:</strong> Systematically looking at the extent of reproducibility of a paper or even in a whole field is important to understand where weaknesses exist. Does, say, economics fall flat while demography shines? How should we approach these reproductions? What aspects contribute to the extent of reproducibility.</li>
<li><strong>Practices of reproducibility:</strong> We need new tools and approaches that encourage us to think more deeply about reproducibility and integrate it into everyday practice.</li>
<li><strong>Teaching reproducibility:</strong> While it is probably too late for most of us, how can we ensure that today’s students don’t repeat our mistakes? What are some case studies that show promise? How can we ensure this doesn’t happen again?</li>
</ul>
<h2 id="section">2022</h2>
<h3 id="timetable">Timetable</h3>
<h4 id="wednesday-23-february-2022">Wednesday, 23 February 2022</h4>
<table>
<colgroup>
<col style="width: 14%" />
<col style="width: 42%" />
<col style="width: 42%" />
</colgroup>
<thead>
<tr class="header">
<th>Time</th>
<th>Speaker</th>
<th>Talk</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>08:40-09:00</td>
<td>Lisa Strug, University of Toronto</td>
<td><a href="https://youtu.be/BEtpZbLTx1E">Introduction and welcome</a></td>
</tr>
<tr class="even">
<td>09:00-09:30</td>
<td>Benjamin Haibe-Kains, University Health Network</td>
<td><a href="https://youtu.be/nmDGgKYtFfU">The (Not-So-)Hard Path To Transparency and Reproducibility in AI Research</a></td>
</tr>
<tr class="odd">
<td>09:30-10:00</td>
<td>Colm-cille Caulfield, University of Cambridge</td>
<td><a href="https://youtu.be/5uiOpc4Rs2w">Reproducibility in an Uncertain World: How should academic data science researchers give advice?</a></td>
</tr>
<tr class="even">
<td>10:00-10:30</td>
<td>Stephen Eglen, University of Cambridge</td>
<td><a href="https://youtu.be/TgDgcqtsFvE">Evaluating the reproducibility of computational results reported in scientific journals</a></td>
</tr>
<tr class="odd">
<td>10:30-11:00</td>
<td>Valentin Danchev, University of Essex</td>
<td><a href="https://youtu.be/wvlSiHJsibk">Reproducibility and Replicability of Large Pre-trained Language Models</a></td>
</tr>
<tr class="even">
<td>11:00-11:30</td>
<td>Monica Alexander, University of Toronto</td>
<td><a href="https://youtu.be/4GCScuiC714">Reproducibility in Demography: where are we at and where can we go?</a></td>
</tr>
<tr class="odd">
<td>11:30-12:00</td>
<td><em>Break</em></td>
<td></td>
</tr>
<tr class="even">
<td>12:00-12:30</td>
<td>Ariel Mundo, University of Arkansas</td>
<td><a href="https://youtu.be/Fvkp20X5xwA">Statistics and reproducibility in biomedical research: Why we need both</a></td>
</tr>
<tr class="odd">
<td>12:30-13:00</td>
<td>Shilaan Alzahawi, Stanford University</td>
<td><a href="https://youtu.be/jynbJjTI10o">Lay perceptions of scientific findings: Swayed by the crowd?</a></td>
</tr>
<tr class="even">
<td>13:00-13:30</td>
<td><em>Break</em></td>
<td></td>
</tr>
<tr class="odd">
<td>13:30-14:00</td>
<td>Fernando Hoces de la Guardia, University of California, Berkeley</td>
<td><a href="https://youtu.be/H9ZyTndcN2o">Social Sciences Reproducibility Platform</a></td>
</tr>
<tr class="even">
<td>14:00-15:30</td>
<td><em>Break</em></td>
<td></td>
</tr>
<tr class="odd">
<td>15:30-16:00</td>
<td>Carl Laflamme, YCharOS</td>
<td><a href="https://youtu.be/uxdi7I7ym_8">Antibody Characterization through Open Science (YCharOS)</a></td>
</tr>
<tr class="even">
<td>16:00-16:30</td>
<td>Robert Hanisch, National Institute of Standards and Technology and Research Data Alliance</td>
<td><a href="https://youtu.be/A8GvDEVrN1Y">Reproducibility: A Metrology Perspective</a></td>
</tr>
<tr class="odd">
<td>16:30-17:00</td>
<td>Yann Joly, McGill University</td>
<td><a href="https://youtu.be/DFMjEcZCctw">Incentivizing open data sharing - what’s in it for me!?</a></td>
</tr>
</tbody>
</table>
<h4 id="thursday-24-february-2022">Thursday, 24 February 2022</h4>
<table>
<colgroup>
<col style="width: 14%" />
<col style="width: 42%" />
<col style="width: 42%" />
</colgroup>
<thead>
<tr class="header">
<th>Time</th>
<th>Speaker</th>
<th>Talk</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>08:30-09:00</td>
<td>Julien Chiquet, Université Paris-Saclay</td>
<td><a href="https://youtu.be/j6P6PNxLbyU">Computo: a journal of the French Statistical Society promoting reproductibility</a></td>
</tr>
<tr class="even">
<td>09:00-09:30</td>
<td>Nick Radcliffe, Global Open Finance Centre at the University of Edinburgh</td>
<td><a href="https://youtu.be/qx_n8mPLpnw">Gentest: Automatic Test Generation for Data Science</a></td>
</tr>
<tr class="odd">
<td>09:30-10:00</td>
<td>Markus Fritsch, University of Passau</td>
<td><a href="https://youtu.be/6EAI-iSQSa4">Towards reproducible GMM estimation</a></td>
</tr>
<tr class="even">
<td>10:00-10:30</td>
<td><em>Break</em></td>
<td></td>
</tr>
<tr class="odd">
<td>10:30-11:00</td>
<td>Aneta Piekut, Sheffield Methods Institute, University of Sheffield</td>
<td><a href="https://youtu.be/aItA2omGVNU">Integrating reproducibility into the curriculum of an undergraduate social sciences degree</a></td>
</tr>
<tr class="even">
<td>11:00-12:30</td>
<td><em>Break</em></td>
<td></td>
</tr>
<tr class="odd">
<td>12:30-13:00</td>
<td>Jason Hattrick,-Simpers, University of Toronto</td>
<td><a href="https://youtu.be/2hc1wVrf0Co">Towards Trust and Reproducibility in Materials AI</a></td>
</tr>
<tr class="even">
<td>13:00-13:30</td>
<td>Aya Mitani, University of Toronto</td>
<td><a href="https://youtu.be/z5wcxdzEisA">Reproducible, reliable, replicable? In-class exercise using peer-reviewed studies</a></td>
</tr>
<tr class="odd">
<td>13:30-14:00</td>
<td>Shannon Ellis, UC San Diego</td>
<td><a href="https://youtu.be/oG6YXIuUny0">Structuring &amp; Managing Group Projects in Large-Enrollment Undergraduate Data Science Courses</a></td>
</tr>
<tr class="even">
<td>14:00-14:30</td>
<td>Maria Tackett, Duke University</td>
<td><a href="https://youtu.be/liw0hHoQzJg">Knit, Commit, and Push: Teaching version control in undergraduate statistics courses</a></td>
</tr>
<tr class="odd">
<td>14:30-15:00</td>
<td><em>Break</em></td>
<td></td>
</tr>
<tr class="even">
<td>15:00-15:30</td>
<td>Lars Vilhuber, Cornell University</td>
<td><a href="https://youtu.be/5wglRf0wc0M">Teaching for large-scale Reproducibility Verification</a></td>
</tr>
<tr class="odd">
<td>15:30-16:00</td>
<td>Michael Geuenich, Lunenfeld Tanenbaum Research Institute and University of Toronto</td>
<td><a href="https://youtu.be/H5lxs5Pm6mU">With great data come great pipelines: creating flexible standardized pipelines for common biomedical analysis tasks using Snakemake</a></td>
</tr>
<tr class="even">
<td>16:00-16:30</td>
<td>Paraskevi Massara, University of Toronto</td>
<td><a href="https://youtu.be/XRBUboGOKH4">MOSS4Research: A maturity model to evaluate and improve reproducibility in research projects</a></td>
</tr>
<tr class="odd">
<td>16:30-17:00</td>
<td>Chris Kenny, Harvard University</td>
<td><a href="https://youtu.be/4skw2JnhUK0">Reproducible Redistricting</a></td>
</tr>
<tr class="even">
<td>17:00-17:30</td>
<td>Dewi Amaliah, Monash University</td>
<td><a href="https://youtu.be/UV_mSj4hKqM">Reproducible Practice in Taming the Wild Data</a></td>
</tr>
</tbody>
</table>
<h4 id="friday-25-february-2022">Friday, 25 February 2022</h4>
<table>
<colgroup>
<col style="width: 14%" />
<col style="width: 42%" />
<col style="width: 42%" />
</colgroup>
<thead>
<tr class="header">
<th>Time</th>
<th>Speaker</th>
<th>Talk</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>09:00-09:30</td>
<td>Marco Prado, University of Western Ontario</td>
<td><a href="https://youtu.be/j5HWsUJQ7SQ">Reproducibility for Behavior Experiments in Basic Science</a></td>
</tr>
<tr class="even">
<td>09:30-10:00</td>
<td>David Grubbs and Lara Spieker, CRC Press</td>
<td><a href="https://youtu.be/trgW7e52LKc">On book publishing</a></td>
</tr>
<tr class="odd">
<td>10:00-11:00</td>
<td>Joelle Pineau, McGill University &amp; Meta (Facebook) AI Research</td>
<td><a href="https://youtu.be/e9CujtFbmmQ">Improving Reproducibility in Machine Learning Research</a></td>
</tr>
<tr class="even">
<td>11:00-11:30</td>
<td>Debbie Yuster, Ramapo College of New Jersey</td>
<td><a href="https://youtu.be/Uz5bM71znIY">Infusing Reproducibility into Introductory Data Science</a></td>
</tr>
<tr class="odd">
<td>11:30-12:00</td>
<td>Colin Rundel, Duke University</td>
<td><a href="https://youtu.be/5Zvs_qRyKzY">Teaching Statistical computing with Git and GitHub</a></td>
</tr>
<tr class="even">
<td>12:00-12:30</td>
<td>Mine Çetinkaya,-Rundel, Duke University and RStudio</td>
<td><a href="https://youtu.be/6p4vOKS6Xls">Reproducible authoring with Quarto</a></td>
</tr>
<tr class="odd">
<td>12:30-13:00</td>
<td>Erin Heerey, Western University</td>
<td><a href="https://youtu.be/ES0FHJHL2LE">The Experimenter in the Room</a></td>
</tr>
<tr class="even">
<td>13:00-13:30</td>
<td>John McLevey, University of Waterloo</td>
<td><a href="https://youtu.be/t9-nhtD8-QA">Reproducibility and Principled Data Processing in Python</a></td>
</tr>
<tr class="odd">
<td>13:30-14:00</td>
<td><em>Break</em></td>
<td></td>
</tr>
<tr class="even">
<td>14:00-14:30</td>
<td>Kevin Wilson, Brown University and Jake Bowers, University of Illinois at Urbana-Champaign</td>
<td><a href="https://youtu.be/kUdlvaC7lW0">Six Tips for Reproducible Field Experiments</a></td>
</tr>
<tr class="odd">
<td>14:30-15:00</td>
<td>Abel Brodeur, University of Ottawa</td>
<td><a href="https://youtu.be/oHo_9E_aid4">Introducing the Institute for Replication</a></td>
</tr>
<tr class="even">
<td>15:00-15:30</td>
<td>Allison Koenecke, Cornell University and Microsoft Research</td>
<td><a href="https://youtu.be/aDJOAN1t1Wg">Reproducible Retrospective Analysis</a></td>
</tr>
<tr class="odd">
<td>15:30-16:30</td>
<td>Michael Hoffman, University Health Network and University of Toronto</td>
<td><a href="https://youtu.be/iDiwLwh1B-c">Reproducibility standards for machine learning in the life sciences</a></td>
</tr>
</tbody>
</table>
<h3 id="presenter-biographies-and-abstracts">Presenter biographies and abstracts</h3>
<h4 id="keynotes">Keynotes</h4>
<ul>
<li><a href="https://www.cs.mcgill.ca/~jpineau/">Joelle Pineau</a>
<ul>
<li>Title: Improving Reproducibility in Machine Learning Research Findings from the NeurIPS Reproduciblity Program and the ML Reproducibility Challenge</li>
<li>Biography: Joelle Pineau is an Associate Professor and William Dawson Scholar at the School of Computer Science at McGill University, where she co-directs the Reasoning and Learning Lab. She is a core academic member of Mila and a Canada CIFAR AI chairholder. She is also co-Managing Director of Facebook AI Research. She holds a BASc in Engineering from the University of Waterloo, and an MSc and PhD in Robotics from Carnegie Mellon University. Dr. Pineau’s research focuses on developing new models and algorithms for planning and learning in complex partially-observable domains. She also works on applying these algorithms to complex problems in robotics, health care, games and conversational agents. She serves on the editorial board of the Journal of Machine Learning Research and is Past-President of the International Machine Learning Society. She is a recipient of NSERC’s E.W.R. Steacie Memorial Fellowship (2018), a Fellow of the Association for the Advancement of Artificial Intelligence (AAAI), a Senior Fellow of the Canadian Institute for Advanced Research (CIFAR), a member of the College of New Scholars, Artists and Scientists by the Royal Society of Canada, and a 2019 recipient of the Governor General’s Innovation Awards.</li>
</ul></li>
<li><a href="https://hoffmanlab.org/">Michael Hoffman</a>
<ul>
<li>Title: Reproducibility standards for machine learning in the life sciences</li>
<li>Abstract: To make machine-learning analyses in the life sciences more computationally reproducible, we propose standards based on data, model and code publication, programming best practices and workflow automation. By meeting these standards, the community of researchers applying machine-learning methods in the life sciences can ensure that their analyses are worthy of trust.</li>
<li>Biography: Michael Hoffman creates predictive computational models to understand interactions between genome, epigenome, and phenotype in human cancers. His influential machine learning approaches have reshaped researchers’ analysis of gene regulation. These approaches include the genome annotation method Segway, which enables simple interpretation of multivariate genomic data. He is a Senior Scientist at Princess Margaret Cancer Centre and Associate Professor in the Departments of Medical Biophysics and Computer Science, University of Toronto. He was named a CIHR New Investigator and has received several awards for his academic work, including the NIH K99/R00 Pathway to Independence Award, and the Ontario Early Researcher Award.</li>
</ul></li>
</ul>
<h4 id="invited-talks">Invited talks</h4>
<ul>
<li><a href="https://sites.google.com/site/abelbrodeur/">Abel Brodeur</a>
<ul>
<li>Title: Introducing the <a href="https://i4replication.org/">Institute for Replication</a></li>
<li>Biography: Abel Brodeur is an associate professor in the department of economics at the University of Ottawa. He is the chair of the Institute for Replication (I4R), which he founded in January 2022. I4R works to improve the credibility of science by systematically reproducing and replicating research findings in leading academic journals.</li>
</ul></li>
<li><a href="http://infosci.cornell.edu/~koenecke/">Allison Koenecke</a>
<ul>
<li>Title: Reproducible Retrospective Analysis</li>
<li>Biography: Allison Koenecke is a postdoc at Microsoft Research in the Machine Learning and Statistics group, and starting Summer 2022 will be an Assistant Professor of Information Science at Cornell University. Her research primarily spans two domains: algorithmic fairness in online services, and causal inference in public health. Previously, she received her PhD from Stanford’s Institute for Computational &amp; Mathematical Engineering.</li>
</ul></li>
<li><a href="https://www.sheffield.ac.uk/smi/people/academic/aneta-piekut">Aneta Piekut</a>
<ul>
<li>Title: Integrating reproducibility into the curriculum of an undergraduate social sciences degree</li>
<li>Abstract: While appreciation for reproducibility and research transparency in social sciences research has grown substantially recently, teaching research reproducibility is still less common, especially at the undergraduate level. Crucially, teaching reproducible research to undergraduate students requires sequencing various open science skills across the curriculum and normalising reproducible research for students. In the talk I will discuss a reproducibility assignment implemented in an undergraduate-level advanced Quantitative Social Sciences course. As part of the assignment, students reproduced a model in a paper published in a high-impact social science journal, added a small extension, and published it as a reproducible report online. I will reflect on the lessons learnt from teaching several interactions of the module and whether one stand-alone ‘replication project’ module is enough to change students’ practice.</li>
<li>Biography: Sociologist specialising in migration and ethnic studies, including measurement of attitudes, migrant integration and segregation. At Sheffield Methods Institute, University of Sheffield, Aneta provides training to undergraduate and postgraduate students in advanced quantitative methods, survey methodology and mixed-method methodology. Aneta is committed to teaching reproducible research methods; in 2020 she was Project TIER Fellow (<a href="https://www.projecttier.org/" class="uri">https://www.projecttier.org/</a>), and in 2021 joined its Executive Committee.</li>
</ul></li>
<li><a href="aimundo.rbind.io">Ariel Mundo</a>
<ul>
<li>Title: Statistics and reproducibility in biomedical research: Why we need both<br />
</li>
<li>Abstract: The biomedical field still struggles at large to make research reproducible. In this talk, I argue that part of this problem is that most of us in biomedical research do not seem to realize the importance of choosing appropriate Statistical models for our data, and how this in turn enables reproducibility. Moreover, I also argue that we need a “statistical rethinking” in biomedical research in order to establish reproducibility as a core aspect of our work.</li>
<li>Biography: Ariel Mundo is a Fulbright alum and PhD Candidate in the Department of Biomedical Engineering at the University of Arkansas. His work focuses on the longitudinal study of changes in cancer metabolism using optical and molecular tools, and the use of semi-parametric methods to analyze such data. He is also an R enthusiast and avid reader.</li>
</ul></li>
<li><a href="https://ayamitani.github.io/">Aya Mitani</a>
<ul>
<li>Title: Reproducible, reliable, replicable? In-class exercise using peer-reviewed studies<br />
</li>
<li>Abstract: I will share my experience in preparing and implementing an in-class exercise to reproduce the results from peer-reviewed publications in health science journals. The course, titled Analysis of Correlated Data, enrolls 20 students mostly pursuing a Master of Science degree in biostatistics. Challenges include finding a suitable clustered or longitudinal study that provides original data and translating the information given (and not given) in the “Methods” section into actual code. Through this exercise, students learn whether the results are not only reproducible but reliable, and whether the analysis can be replicated on a different set of data. The goal through this exercise is to teach the students how to write an applied manuscript or report as modern biostatisticians.<br />
</li>
<li>Biography: I am an Assistant Professor in the Division of Biostatistics at the Dalla Lana School of Public Health (DLSPH) of the University of Toronto. I obtained my Ph.D. in Biostatistics from Boston University and did my postdoctoral research fellowship at Harvard T. H. Chan School of Public Health. My research includes the development of statistical methods for complex oral health data, multiple imputation for missing data, modelling agreement in cancer screening studies, and biased sampling designs in surveys and observational studies. At DLSPH, I teach Analysis of Correlated Data and Introduction to Joint Modeling in Health Research. I am passionate about incorporating good reproducible research practices into my teaching. In 2021, I co-founded the Health Data Working Group at DLSPH to provide an accessible space for students and researchers to learn about data and coding outside of the classroom. I live in Etobicoke with my husband and two children.</li>
</ul></li>
<li><a href="http://bhklab.ca">Benjamin Haibe-Kains</a>
<ul>
<li>Title: The (Not-So-)Hard Path To Transparency and Reproducibility in AI Research<br />
</li>
<li>Abstract: As artificial intelligence (AI) becomes a method of choice to analyze biomedical data, the field is facing multiple challenges around research reproducibility and transparency. Given the proliferation of studies investigating the applications of AI in research and clinical studies, it is essential for independent researchers to be able to scrutinize and reproduce the results of a study using its materials, and build upon them in future studies. Computational reproducibility is achievable when the data can easily be shared and the required computational resources are relatively common. However, the complexity of AI algorithms and their implementation, the need for specific computer hardware and the use of sensitive biomedical data represent major obstacles in healthy-related AI research. In this talk, I will describe the various aspects of an AI biomedical study that are necessary for reproducibility and the platforms that exist for sharing these materials with the scientific community.<br />
</li>
<li>Biography: Dr. Benjamin Haibe-Kains is a Senior Scientist at the Princess Margaret Cancer Centre (PM), University Health Network, and Associate Professor in the Medical Biophysics department of the University of Toronto. Dr. Haibe-Kains earned his PhD in Bioinformatics at the Université Libre de Bruxelles (Belgium). Supported by a Fulbright Award, he did his postdoctoral fellowship at the Dana-Farber Cancer Institute and Harvard School of Public Health (USA). Dr. Haibe-Kains’ research focuses on the integration of high-throughput data from various sources to simultaneously analyze multiple facets of carcinogenesis. Dr. Haibe- Kains’ team is analyzing large-scale radiological and (pharmaco)genomic datasets to develop new prognostic and predictive models to improve cancer care.</li>
</ul></li>
<li>Carl Laflamme
<ul>
<li>Title: Antibody Characterization through Open Science (YCharOS)</li>
<li>Abstract: Global sales of commercial antibodies are estimated at $2 billion per year with approximately half that money wasted on underperforming reagents. Both public and private sectors agree that a robust, independent, and scalable process to characterize commercial antibodies is required, but all attempts to find a solution have failed due to the tangle of conflicting interests in both academia and industry. YCharOS (Antibody Characterization using Open Science), in collaboration with the Structural Genomics Consortium (SGC) and the Montreal Neurological Institute (The Neuro, McGill University) has created an open science ecosystem in which antibody manufacturers, knockout cell line providers, academics, pharma and granting agencies contribute resources and knowledge to solve the antibody liability crisis. We have already publicly shared the identification of hundreds of high-performing antibodies for dozens of neuroscience targets. We have scaled up our platform, developed automation and expanded our team. We now aim to characterize antibodies for the human proteome.</li>
</ul></li>
<li><a href="https://www.christophertkenny.com/">Chris Kenny</a>
<ul>
<li>Title: Reproducible Redistricting<br />
</li>
<li>Abstract: Modern redistricting is known for occurring behind closed doors where incumbent politicians can work to advance their co-partisan’s interests. Recent advancements in political science and statistical research have developed the tools to help resolve these problems. I overview the R-package-based workflow that the ALARM Project and its members use for research, advocacy, and testimony to courts. Key packages developed for these purposes include redist, redistmetrics, and geomander.</li>
<li>Biography: Chris Kenny is a Ph.D. candidate in the Department of Government at Harvard University, studying American Politics and Political Methodology. He is currently the Political Science Pre-Doctoral Fellow at the Harvard Election Law Clinic. His substantive focus is on redistricting and gerrymandering. He primarily develops open-source R tools for analyzing redistricting and voting rights in geographic and contemporary contexts. He is an affiliate with the Center for American Political Studies at Harvard University, The Institute for Quantitative Social Science, and the Algorithm-Assisted Redistricting Methodology (ALARM) Project.</li>
</ul></li>
<li><a href="http://www2.stat.duke.edu/~cr173/">Colin Rundel</a>
<ul>
<li>Title: Teaching Statistical computing with Git and GitHub<br />
</li>
</ul></li>
<li>Colm-cille Caulfield
<ul>
<li>Title: Reproducibility in an Uncertain World: How should academic data science researchers give advice? open science-type initiatives<br />
</li>
</ul></li>
<li>David Grubbs and Lara Spieker
<ul>
<li>Title: On book publishing<br />
</li>
<li>Abstract: In this very practical and interactive workshop, four book editors from Chapman and Hall/CRC will discuss why you should consider publishing an R or Data Science book and why you should work with CRC. The editors will go over the publishing process and provide best practices for shaping your ideas and submitting a book proposal; discuss their bestsellers and popular series as well as emerging topics and trends. The lively discussion will provide plenty of opportunities for the attendees to ask questions and discuss ideas.</li>
</ul></li>
<li><a href="https://www.ramapo.edu/tas/faculty/debbie-yuster/">Debbie Yuster</a>
<ul>
<li>Title: Infusing Reproducibility into Introductory Data Science</li>
<li>Abstract: In this talk, I will discuss the role of reproducibility in my Introduction to Data Science course. The course has no prerequisites, so many students are coding and analyzing data for the first time. They develop habits of reproducibility from the start: their analyses are done within R Markdown documents, and GitHub is used to facilitate both version control and collaboration among teammates. Through scaffolded coding exercises, gradual onboarding to GitHub, and focusing on a small subset of GitHub functionality, even beginner students can become adept at using these technologies. I will also discuss tips learned from teaching the course in a fully remote format, and will provide pointers to training resources for instructors who want to use similar tools and workflows in their own courses.</li>
<li>Biography: Debbie Yuster is an Assistant Professor of Data Science and Mathematics at Ramapo College of New Jersey. She holds a Ph.D. in Mathematics from Columbia University. Prior to joining Ramapo, Debbie served as a math professor at SUNY Maritime College, earning the SUNY Chancellor’s Award for Excellence in Teaching. Debbie served as a Visiting Data Science Scholar at the Wall Street Journal, and has cultivated industry partnerships leading to undergraduate research projects. She also has an interest in K-12 STEM outreach, having worked with secondary school teachers and students for many years.</li>
</ul></li>
<li><a href="https://github.com/Dewi-Amaliah">Dewi Amaliah</a>
<ul>
<li>Title: Reproducible Practice in Taming the Wild Data</li>
<li>Abstract: I will talk about my experience in refreshing the wages data from the prominent survey, NLSY79, which is used as an example of longitudinal data in a textbook (Singer and Willet, 2003). The motivation of this study is to demonstrate the steps (extracting, tidying, cleaning, and exploring) and clearly articulate the decision made when data is refreshed from the raw (wild) to the textbook (tame) data. All of those steps are documented to ensure reproducibility.</li>
</ul></li>
<li>Erin Heerey
<ul>
<li>Title: The Experimenter in the Room</li>
</ul></li>
<li>Fernando Hoces de la Guardia
<ul>
<li>Title: Social Sciences Reproducibility Platform Social Sciences Reproducibility Platform<br />
</li>
</ul></li>
<li>Kevin Wilson and <a href="http://www.jakebowers.org/">Jake Bowers</a>
<ul>
<li>Title: Six Tips for Reproducible Field Experiments</li>
</ul></li>
<li><a href="https://mse.utoronto.ca/faculty-staff/professors/jason-hattrick-simpers/">Jason Hattrick-Simpers</a>
<ul>
<li>Title: Towards Trust and Reproducibility in Materials AI</li>
<li>Biography: Jason Hattrick-Simpers is a Professor at the Department of Materials Science and Engineering, University of Toronto, and a Research Scientist at CanmetMATERIALS. He graduated with a B.S. in Mathematics and a B.S. in Physics from Rowan University and a Ph.D. in Materials Science and Engineering from the University of Maryland. Prior to joining UofT Prof. Hattrick-Simpers was a staff scientist at the National Institute of Standards and Technology (NIST) in Gaithersburg, MD where he co-developed tools for discovering novel corrosion resistance of alloys, developed active learning approaches to guide thin film and additive manufacturing alloy studies, and developed tools and best practices to enable trust in AI within the materials science community.</li>
</ul></li>
<li><a href="https://uwaterloo.ca/knowledge-integration/people-profiles/john-mclevey">John McLevey</a>
<ul>
<li>Title: Reproducibility and Principled Data Processing in Python</li>
</ul></li>
<li><a href="https://computo.sfds.asso.fr">Julien Chiquet</a>
<ul>
<li>Title: Computo: a journal of the French Statistical Society promoting reproductibility<br />
</li>
<li>Abstract: This talk will present Computo (<a href="https://computo.sfds.asso.fr/" class="uri">https://computo.sfds.asso.fr/</a>), an academic journal that has just been born, which calls for higher standards in the publication of scientific results. In order to achieve this goal, Computo goes beyond classical static publications by leveraging technical advances in literate programming and scientific reporting. Computo focuses on computational and algorithmic methodological contributions to the field of statistics and machine learning. The journal is designed to allow authors to demonstrate the usefulness of their methods for data analysis, but also to promote the numerical illustration of theoretical properties. In the era of the reproducibility crisis, Computo differs from other journals in the centrality given to the issues of replicability and open science: - Computo is distributed solely online, free for authors and readers; - It systematically makes available the exchanges between authors and reviewers, the latter being able to choose to remain anonymous; - Computo uses an original publication format that guarantees the reproducibility of results: articles are submitted and published in the form of interactive documents (“notebook” integrating text, code, equations and bibliographic references), associated with a github repository configured to demonstrate, dynamically and durably, the reproducibility of the contribution. On the Computo submission page, we offer various templates to prepare your submissions, as well as an example of a finalized article and the associated repository.</li>
<li>Biography: Julien Chiquet, editor of Computo, is a senior researcher in statistical learning. He is supported for this project by co-editors Chloé Azencott, Pierre Neuvial and Nelle Varoquaux, all researchers in machine learning and statistics</li>
</ul></li>
<li><a href="https://www.ilr.cornell.edu/people/lars-vilhuber">Lars Vilhuber</a>
<ul>
<li>Title: Teaching for large-scale Reproducibility Verification<br />
</li>
<li>Abstract: We describe a unique environment in which undergraduate students from various STEM and social science disciplines are trained in data provenance and reproducible methods, and then apply that knowledge to real, conditionally accepted manuscripts and associated replication packages. We describe in detail the recruitment, training, and regular activities. While the activity is not part of a regular curriculum, the skills and knowledge taught through explicit training of reproducible methods and principles, and reinforced through repeated application in a real-life workflow, contribute to the education of these undergraduate students, and prepare them for post-graduation jobs and further studies.</li>
<li>Biography: Lars Vilhuber holds a Ph.D. in Economics from Université de Montréal, Canada, and is currently on the faculty of the Cornell University Economics Department. He has interests in labor economics, statistical disclosure limitation and data dissemination, and reproducibility and replicability in the social sciences. He is the Data Editor of the American Economic Association, and Managing Editor of the Journal of Privacy and Confidentiality.</li>
</ul></li>
<li><a href="https://www.sickkids.ca/en/staff/s/lisa-strug/">Lisa Strug</a>
<ul>
<li>Title: Introduction and overview</li>
<li>Biography: Dr. Strug is Professor in the Departments of Statistical Sciences, Computer Science and cross-appointed in Biostatistics at the University of Toronto and is a Senior Scientist in the Program in Genetics and Genome Biology at the Hospital for Sick Children. Dr. Strug is the inaugural Director of the Data Sciences Institute (DSI), a tri-campus, multi-divisional, multi-institutional, multi-disciplinary hub for data science activity at the University of Toronto and affiliated Research Institutes. The DSI’s goal is to accelerate the impact of data sciences across the disciplines to address pressing societal questions and drive positive social change. Dr. Strug holds several other leadership positions at the University of Toronto including the Director of the Canadian Statistical Sciences Institute Ontario Region (CANSSI Ontario), and at the Hospital for Sick Children as Associate Director of the Centre for Applied Genomics and the Lead of the Canadian Cystic Fibrosis Gene Modifier Consortium and the Biology of Juvenile Myoclonic Epilepsy International Consortium. She is a statistical geneticist and her research focuses on the development of novel statistical approaches to analyze and integrate multi-omics data to identify genetic contributors to complex human disease. She has received several honours including the Tier 1 Canada Research Chair in Genome Data Science.</li>
</ul></li>
<li>Marco Prado
<ul>
<li>Title: Reproducibility for Behavior Experiments in Basic Science</li>
<li>Biography: Marco Prado is scientist at the Robarts Research Institute and a full professor at the University of Western Ontario, where he holds a Canada Research Chair in Neurochemistry of Dementia. He is interested in understanding how neurochemical alterations in neurodegenerative diseases contribute to protein misfolding and cognitive dysfunction. He has made contributions to understanding maladaptive signaling in Alzheimer’s and Prion diseases by investigating physiological functions of the prion protein and in how molecular chaperones affect signaling and protein misfolding in neurodegenerative diseases. He has developed multiple genetic mouse models of neurochemical dysfunction in dementia. Marco’s group combines the use of sophisticated touchscreen tests of high-level cognition and detailed biochemical analysis to reveal several mechanisms regulating executive function and mechanisms of pathological changes in mouse models. He is currently spearheading with several colleagues an Open Science Repository (www.mousebytes.ca) for high-level cognitive data in mouse models of neurodegenerative disease. This effort will support a community of more than 300 laboratories to increase reproducibility and replicability of cognitive datasets in pre-clinical research. Marco Prado received several awards, including the Guggenheim Fellowship, for his work and has published over 200 manuscripts.</li>
</ul></li>
<li><a href="https://www.mariatackett.net/">Maria Tackett</a>
<ul>
<li>Title: Knit, Commit, and Push: Teaching version control in undergraduate statistics courses</li>
<li>Abstract: In recent years there has been increased focus on incorporating the skills required to conduct well-documented and reproducible analyses in the undergraduate statistics curriculum. Because data analysis is an iterative process, version control, a record of changes to a set of files over time, is a foundational part of a reproducible workflow. In this talk, I will describe how version control with Git can be included as a learning objective in the first and second statistics courses. I’ll discuss strategies for introducing version control to students, incorporating it in individual and team-based assignments, and assessing students’ understanding. I’ll also share lessons learned and an example of how this can be implemented using RStudio and GitHub.</li>
<li>Biography: Maria Tackett is an Assistant Professor of the Practice in the Department of Statistical Science at Duke University. Prior to joining the faculty at Duke, Maria earned a Ph.D. in Statistics from the University of Virginia and worked as a statistician at Capital One. Her work focuses on using active learning strategies to increase engagement in large undergraduate statistics courses, and understanding how classroom practices impact students’ sense of community in these courses. Maria is active in the statistics education community, including serving as the current Communications Officer for the ASA Section on Statistics and Data Science Education.</li>
</ul></li>
<li>Markus Fritsch
<ul>
<li>Title: Towards reproducible GMM estimation<br />
</li>
<li>Abstract: Generalized method of moments (GMM) estimation is a way forward in regression setups where endogeneity is present. A practically relevant area of application is the estimation of linear dynamic panel data models. This context forces the researcher to make many decisions that seem marginal at first, but which often affect the estimation and inference dramatically. The decisions comprise the number and type of employed moment conditions, their weighting scheme, how covariates and/or dummy variables are included, whether we iterate the estimation procedure and/or bias-correct, etc. Due to the many possible choices, clear documentation and reproducibilty are vital for the communication of GMM estimation results. We provide guidelines for reproducible GMM estimation and demonstrate their relevance by replicating and extending several empirical applications.<br />
</li>
<li>Biography: Markus Fritsch is Assistant Professor at the Chair of Statistics and Data Analytics of the University of Passau. He is the creator of the CRAN package pdynmc. His research interest include Data Science &amp; Statistical Learning, GMM estimation, Quantile regression, and reproducible Applied Statistics.</li>
</ul></li>
<li>Michael Geuenich
<ul>
<li>Title: With great data come great pipelines: creating flexible standardized pipelines for common biomedical analysis tasks using Snakemake<br />
</li>
<li>Abstract: Biomedical data analysis pipelines are becoming increasingly complex as projects frequently involve the analysis of raw data from distinct batches and experimental modalities. Work frequently starts with processing and normalizing several large datasets in a variety of ways, often requiring custom filtering approaches for each individual dataset. Existing and novel analysis methods are then frequently applied to the processed data using a variety of parameters prior to subsequent benchmarking, resulting in many individual analysis steps that need to be tied together. Importantly, some data processing steps are frequently dependent on the data itself, requiring inspection of preliminary results before being able to run a standardized pipeline in full. In addition, pre-processing steps are frequently revised as part of the iterative analysis workflow common to most projects, thus requiring downstream analyses to be re-run as input data changes. These challenges make it cumbersome and error prone to run individual analysis steps manually. Workflow managers such as Snakemake allow for the creation of reproducible and easily</li>
<li>Biography: Michael is a PhD student in the computational track of the molecular genetics department at the University of Toronto and at the Lunenfeld Tanenbaum Research Institute with Kieran Campbell. His work focusses on better understanding immune escape in pancreatic cancer using machine learning tools and a diverse set of -omics data.</li>
</ul></li>
<li><a href="https://mine-cr.com/">Mine Çetinkaya-Rundel</a>
<ul>
<li>Title: Reproducible authoring with Quarto</li>
<li>Biography: I am a Professor of the Practice and the Director of Undergraduate Studies at the Department of Statistical Science and an affiliated faculty in the Computational Media, Arts, and Cultures program at Duke University. My work focuses on innovation in statistics and data science pedagogy, with an emphasis on computing, reproducible research, student-centered learning, and open-source education. I work on integrating computation into the undergraduate statistics curriculum, using reproducible research methodologies and analysis of real and complex datasets. In addition to my academic position, I also work with RStudio, where I focus primarily on education for open-source R packages as well as building resources and tools for educators teaching statistics and data science with R and RStudio.</li>
</ul></li>
<li><a href="https://www.monicaalexander.com/">Monica Alexander</a>
<ul>
<li>Title: Reproducibility in Demography: where are we at and where can we go?<br />
</li>
<li>Biography: Monica Alexander is an Assistant Professor in Statistical Sciences and Sociology at the University of Toronto. Her research focuses on developing statistical methods to help measure demographic and health outcomes. She received a PhD in Demography and Masters in Statistics from the University of California, Berkeley. She has worked on research projects with organizations such as UNICEF, the World Health Organization, the Bill and Melinda Gates Foundation, and the Human Mortality Database.</li>
</ul></li>
<li><a href="https://www.stochasticsolutions.com/about/">Nick Radcliffe</a>
<ul>
<li>Title: Gentest: Automatic Test Generation for Data Science<br />
</li>
<li>Abstract: This talk will focus on reference tests—scripts that test the ongoing correctness of scripts, programs and pipelines with a particular focus on data science-oriented tasks. The TDDA library has long offered support to allow humans to write useful tests for data science workflows, with a focus on supporting tests for what might be called semantic/functional correctness, rather than syntactic/form correctness. New “Gentest” functionality in TDDA goes further by automating large parts of test production. Using Gentest, researchers can concentrate on developing robust/correct analysis pipelines, verifying them in the usual way (probably by hand), and then use Gentest to generate executable tests automatically. Although Gentest is written in Python, it can be also used to generate tests for R or almost any other language. If all goes well, this talk will include a demonstration of automatically generating tests for R scripts.<br />
</li>
<li>Biography: Nick Radcliffe is the founder of the data science consulting and software firm, Stochastic Solutions Limited, the Interim Chief Scientist at the Global Open Finance Centre of Excellence, and a Visiting Professor in Maths and Stats at University of Edinburgh, Scotland. His background combines theoretical physics, operations research, machine learning and stochastic optimization. Nick’s current research interests include a focus on test-driven data analysis, (an approach to improving correctness of analytical results that combines ideas from reproducible research and test-driven development) and privacy-respecting analysis. He is the lead author of the open-source Python tdda package, which provides practical tools for testing analytical software and data, and also of the Miró data analysis suite.</li>
</ul></li>
<li>Paraskevi Massara
<ul>
<li>Title: MOSS4Research: A maturity model to evaluate and improve reproducibility in research projects.<br />
</li>
<li>Abstract: Our ability to gather large amounts of data, store it and analyze it efficiently has created new research opportunities in health sciences and it has led to novel practices. One such practice is the creation of large datasets that can be used in multiple studies effectively increasing our research output. However, big data is no free lunch and it comes with its own challenges. On one hand, improper management of data may lead to problems when communicating or sharing data. Different terminology, inaccessible storage, ethical/economic/social barriers may be some of the problems related to sharing the common large datasets. On the other hand, improper management is not constrained only in data, but can extend to the analysis as well, where processes or analytical tasks are not properly documented or permanently stored. These problems significantly inhibit the reproducibility of studies, which in turn may make the verification of research results practically impossible, and they can also lead to waste in terms of lost data, time, effort and funds. Other practical domains, such as computer science or engineering, have long employed methods to systematically document data and processing tasks to allow for repetition and reproducibility. Based on such methods, we propose a novel framework to evaluate the maturity of the reproducibility practices employed in the context of individual projects or within an entire research team. The framework consists of a self-assessment questionnaire and a maturity model to allow teams to evaluate the maturity of their responsibility practices and a guide on how to increase their maturity level. The guide contains practices drawn from other domains to improve communication, collaboration and reproducibility.<br />
</li>
<li>Biography: Paraskevi Massara is a PhD candidate supervised by Drs. Elena Comelli and Robert Bandsma. Her research interests include growth pattern detection in children in association with gut microbiome. She is a coding enthusiast and an aspiring data science have extensive practical experience with programming, machine learning and statistics, and development and management platforms such as Github. She is a member of R ladies and Women Who Code. She is the recipient among others of Ontario Graduate Scholarship, Peterborough Hunter Charitable Foundation Graduate Award, Connaught International Scholarship.</li>
</ul></li>
<li><a href="https://www.nist.gov/people/robert-hanisch">Robert Hanisch</a>
<ul>
<li>Title: Reproducibility: A Metrology Perspective</li>
<li>Biography: Dr. Robert J. Hanisch is the Director of the Office of Data and Informatics in the Material Measurement Laboratory at NIST. Prior to this appointment (July 2014) he was a Senior Scientist at the Space Telescope Science Institute (STScI), Baltimore, Maryland, and Director of the US Virtual Astronomical Observatory. In the past twenty-five years Dr. Hanisch has led many efforts in the astronomy community in the area of information systems and services, focusing particularly on efforts to improve the accessibility and interoperability of data archives and catalogs. He was the first chair of the International Virtual Observatory Alliance Executive Committee (2002-2003) and continues as a member of the IVOA Executive. From 2000 to 2002 he served as Chief Information Officer at STScI, overseeing all computing, networking, and information services for the Institute. Prior to that he had oversight responsibilities for the Hubble Space Telescope Data Archive and led the effort to establish the Multimission Archive at Space Telescope—MAST—as the optical/UV archive center for NASA astrophysics missions. He has served as chair of the Program Organizing Committee for the Astronomical Data Analysis Software and Systems (ADASS) conferences, chair of the Astrophysics Data Centers Coordinating Committee, and co-chair of the Decadal Survey Study Group on Computation, Simulation, and Data Handling. He is currently president of IAU Commission 5 (Data and Documentation), chair of the IAU Comm. 5 Working Group on Virtual Observatories, Data Centers, and Networks, and co-chair of the Comm. 5 Working Group on Libraries. He completed his Ph.D. in Astronomy in 1981 at the University of Maryland, College Park, working in the field of extragalactic radio astronomy with Prof. William Erickson.</li>
</ul></li>
<li><a href="http://www.shanellis.com/">Shannon Ellis</a>
<ul>
<li>Title: Structuring &amp; Managing Group Projects in Large-Enrollment Undergraduate Data Science Courses</li>
<li>Abstract: Computational notebooks are a popular tool for generating technical data science reports, as they allow for narrative text, code, and code outputs in a single explanatory document. Given their popularity, many data science courses utilize computational notebooks for instruction, assignments, and projects, the output of which can be analyzed to better understand student behavior and improve instruction. Here, we present the results from the analysis of 686 final group data science projects from 8 iteractions of the undergraduate course COGS 108 Data Science in Practice to explain how students approach open-ended data science projects and provide data science instructors with general recommendations on structuring and managing reproducible data science projects in large-enrollment data science courses.<br />
</li>
<li>Biography: Shannon E. Ellis is an Assistant Teaching Professor at UC San Diego in the Cognitive Science Department, where her primary focus is teaching programming and data science to thousands of undergraduate students each academic year. Prior to her arrival at UC San Diego, Shannon received her Ph.D. in Human Genetics from the Johns Hopkins School of Medicine and completed a postdoctoral fellow in the Department of Biostatistics at the Johns Hopkins Bloomberg School of Public Health. Shannon is particularly passionate about data science, ethical data analysis, and education. She aims to ensure that data science education is accessible to everyone, with a particular focus on individuals from marginalized groups who typically have not had access to such materials and training.</li>
</ul></li>
<li><a href="https://shilaan.rbind.io/">Shilaan Alzahawi</a>
<ul>
<li>Title: Lay perceptions of scientific findings: Swayed by the crowd?</li>
<li>Abstract: Every day, important scientific findings are rejected at large. To increase public faith in science, some have proposed the use of crowd science. Drawing from theories on social norms and numerical cognition, we test whether crowd science improves lay perceptions of scientific findings. We run an experiment (N = 2,019; preregistration, data, code, and materials at osf.io/vedb4) to study the effects of scientific findings emerging from a crowd of researchers (vs. a typical research collaboration) on lay consumers’ posterior beliefs, confidence in an aggregate effect size estimate, and ratings of credibility, bias, and error. We focus on crowdsourced data analysis: a crowd of scientists who independently analyze the same data to estimate and report a parameter of interest. Contrary to our hypotheses, we do not find that consistent crowd estimates increase the sway and credibility of scientific findings to lay consumers: instead, to our surprise, they lead to lower posterior beliefs and higher ratings of error. In the future, it is important for crowd scientists to consider how to tackle science skepticism and effectively communicate variable crowd estimates to lay consumers.<br />
</li>
<li>Biography: Shilaan Alzahawi is a Master student in Statistics at Ghent University and a PhD candidate in Organizational Behavior at Stanford University. Shilaan is interested in meta-science and inferential statistics, with a particular interest in the coordination and effectiveness of large-scale science collaborations.</li>
</ul></li>
<li>Stephen Eglen
<ul>
<li>Title: Evaluating the reproducibility of computational results reported in scientific journals</li>
<li>Abstract: A recent study (&lt; <a href="http://dx.doi.org/10.1371/journal.pbio.3001107" class="uri">http://dx.doi.org/10.1371/journal.pbio.3001107</a>&gt;) estimated that only 2% of biomedical articles shared code relating to computations. This lack of sharing of code inhibits reproducibility of findings and reusability of methods. I will introduce our CODECHECK project &lt; <a href="https://codecheck.org.uk" class="uri">https://codecheck.org.uk</a>&gt; that reviews computational findings underlying research articles in biosciences. Compared to traditional peer review, this review is open and interactive, with the aim of helping all authors make their work reproducible. All code/data required to reproduce computational results, and the results themselves, are shared freely following FAIR guidelines. We hope our system will be used across multiple publishers and bring a cultural change towards more transparent, open, and reusable computational workflows. This is joint work with Daniel Nust.</li>
<li>Biography: SJE is Professor of Computational Neuroscience, in the Department of Applied Mathematics and Theoretical Physics, University of Cambridge. He has a long-standing interest in open science and reproducible research. He co-leads the CODECHECK project for reproducibility of computations in scientific publications (<a href="https://codecheck.org.uk" class="uri">https://codecheck.org.uk</a>). He is an associate editor for BiorXiv and is on advisory boards for F1000 Research and Gigabyte.</li>
</ul></li>
<li><a href="https://www.essex.ac.uk/people/danch45708/valentin-danchev">Valentin Danchev</a>
<ul>
<li>Title: Reproducibility and Replicability of Large Pre-trained Language Models</li>
<li>Abstract: A major recent development in artificial intelligence and deep learning research are large language models (LLMs) (e.g., BERT, GPT-3, Gopher) that are trained on a massive amount of language data and are subsequently applied to a wide range of downstream tasks. Over the last couple of years, LLMs have been adopted and have shown promise across research domains, pointing to the importance of evaluating the scientific potential and challenges of these models through the lenses of research transparency, computational reproducibility, and replicability. While challenges for reproducibility and replicability in data-intensive computational applications are not new, pre-trained LLMs built on deep learning approaches bring some novel epistemic challenges as well as related ethical and social risks. Specifically, the massive and often sensitive, publicly unavailable, and proprietary data sets on which these models are pre-trained; the scale of the models with hundreds of billions of parameters and associated computationally intensive infrastructure; and the pre-trained nature of the models forming a basis for subsequent applications in the context of restricted access to many of the models, their software, and training procedures can all pose challenges to research transparency, computational reproducibility, and replicability. I will discuss these challenges and outline possible improvements drawing on principles of responsible and reproducible research and on recent frameworks and practices in data-intensive computational sciences aiming to securely access and model sensitive data at scale.<br />
</li>
<li>Biography: Valentin Danchev is a Lecturer in Computational Social Science at the University of Essex and a Fellow of the Software Sustainability Institute. He holds a DPhil from the University of Oxford and held postdoctoral positions at the University of Chicago and the Stanford University School of Medicine. His research combines computational methods from data science and network analysis with approaches from reproducible research and metascience to study the transparency, reproducibility, bias, and social impact of data-intensive research, with a current focus on evaluating and improving the transparency and reproducibility of applications of data science, artificial inteligence, and machine learning in the social and health sciences. In another stream of research, he uses computational social science and network analysis to examine health-related misinformation, digital-health interventions, and inequality in network structures of global migration. He teaches data science with an emphasis on open reproducible workflows and responsible analysis of real-world data.</li>
</ul></li>
<li><a href="https://yannjoly.openum.ca/">Yann Joly</a>
<ul>
<li>Title: Incentivizing open data sharing - what’s in it for me!?</li>
</ul></li>
</ul>
<h2 id="section-1">2021</h2>
<h3 id="timetable-1">Timetable</h3>
<h4 id="thursday-25-february-2021">Thursday, 25 February, 2021</h4>
<table>
<colgroup>
<col style="width: 16%" />
<col style="width: 45%" />
<col style="width: 18%" />
<col style="width: 18%" />
</colgroup>
<thead>
<tr class="header">
<th>Time</th>
<th>Speaker</th>
<th>Focus</th>
<th>Recording</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>9:00-9:10am</td>
<td>Rohan Alexander, <em>University of Toronto</em></td>
<td>Welcome</td>
<td>-</td>
</tr>
<tr class="even">
<td>9:10-9:20am</td>
<td>Radu Craiu, <em>University of Toronto</em></td>
<td>Opening remarks</td>
<td><a href="https://youtu.be/JGGVEgMBURU" class="uri">https://youtu.be/JGGVEgMBURU</a></td>
</tr>
<tr class="odd">
<td>9:20-9:30am</td>
<td>Wendy Duff, <em>University of Toronto</em></td>
<td>Opening remarks</td>
<td><a href="https://youtu.be/Z3aWU1A0FCw" class="uri">https://youtu.be/Z3aWU1A0FCw</a></td>
</tr>
<tr class="even">
<td>9:30-10:25am</td>
<td>Mine Çetinkaya-Rundel, <em>University of Edinburgh</em></td>
<td>Keynote - Teaching</td>
<td><a href="https://youtu.be/ANH2tv2vkew" class="uri">https://youtu.be/ANH2tv2vkew</a></td>
</tr>
<tr class="odd">
<td>10:30-11:30am</td>
<td>Riana Minocher, <em>Max Planck Institute for Evolutionary Anthropology</em></td>
<td>Keynote - Evaluating</td>
<td><a href="https://youtu.be/O3t8TwWeli0" class="uri">https://youtu.be/O3t8TwWeli0</a></td>
</tr>
<tr class="even">
<td>11:30-11:55am</td>
<td>Tiffany Timbers, <em>University of British Columbia</em></td>
<td>Teaching</td>
<td><a href="https://youtu.be/mh93W8XimOg" class="uri">https://youtu.be/mh93W8XimOg</a></td>
</tr>
<tr class="odd">
<td>Noon-12:25pm</td>
<td>Tyler Girard, <em>University of Western Ontario</em></td>
<td>Teaching</td>
<td><a href="https://youtu.be/k3qgmUAjIvA" class="uri">https://youtu.be/k3qgmUAjIvA</a></td>
</tr>
<tr class="even">
<td>12:30-12:55pm</td>
<td>Shiro Kuriwaki, <em>Harvard University</em></td>
<td>Practices</td>
<td><a href="https://youtu.be/-J-eiPnmoNE" class="uri">https://youtu.be/-J-eiPnmoNE</a></td>
</tr>
<tr class="odd">
<td>1:00-1:25pm</td>
<td>Meghan Hoyer, <em>Washington Post</em> &amp; Larry Fenn <em>AP</em></td>
<td>Practices</td>
<td><a href="https://youtu.be/FFwMfNk83rc" class="uri">https://youtu.be/FFwMfNk83rc</a></td>
</tr>
<tr class="even">
<td>1:30-1:55pm</td>
<td>Tom Barton, <em>Royal Holloway, University of London</em></td>
<td>Evaluating</td>
<td><a href="https://youtu.be/YTdhcSDqFNQ" class="uri">https://youtu.be/YTdhcSDqFNQ</a></td>
</tr>
<tr class="odd">
<td>2:00-2:25pm</td>
<td><em>Break</em></td>
<td>-</td>
<td>-</td>
</tr>
<tr class="even">
<td>2:30-2:55pm</td>
<td>Mauricio Vargas, <em>Catholic University of Chile</em> &amp; Nicolas Didier <em>Arizona State University</em></td>
<td>Evaluating</td>
<td><a href="https://youtu.be/VpTavLYEMgg" class="uri">https://youtu.be/VpTavLYEMgg</a></td>
</tr>
<tr class="odd">
<td>3:00-3:25pm</td>
<td>Jake Bowers, <em>University of Illinois &amp; The Policy Lab</em></td>
<td>Practices</td>
<td><a href="https://youtu.be/3N0YwJIbbHg" class="uri">https://youtu.be/3N0YwJIbbHg</a></td>
</tr>
<tr class="even">
<td>3:30-3:55pm</td>
<td>Amber Simpson, <em>Queens University</em></td>
<td>Practices</td>
<td><a href="https://youtu.be/uUfrcB6aynQ" class="uri">https://youtu.be/uUfrcB6aynQ</a></td>
</tr>
<tr class="odd">
<td>4:00-4:25pm</td>
<td>Garret Christensen, <em>US FDIC</em></td>
<td>Evaluating</td>
<td><a href="https://youtu.be/595KkVKJ29w" class="uri">https://youtu.be/595KkVKJ29w</a></td>
</tr>
<tr class="even">
<td>4:30-4:55pm</td>
<td>Yanbo Tang, <em>University of Toronto</em></td>
<td>Practices</td>
<td><a href="https://youtu.be/0x6gOkldOvk" class="uri">https://youtu.be/0x6gOkldOvk</a></td>
</tr>
<tr class="odd">
<td>5:00-5:25pm</td>
<td>Lauren Kennedy, <em>Monash University</em></td>
<td>Practices</td>
<td><a href="https://youtu.be/HhfogRbgbA4" class="uri">https://youtu.be/HhfogRbgbA4</a></td>
</tr>
<tr class="even">
<td>5:30-6:00pm</td>
<td>Lisa Strug, <em>University of Toronto &amp; CANSSI Ontario</em></td>
<td>Closing remarks</td>
<td><a href="https://youtu.be/B_9puTSp3f8" class="uri">https://youtu.be/B_9puTSp3f8</a></td>
</tr>
</tbody>
</table>
<h4 id="friday-26-february-2021">Friday, 26 February, 2021</h4>
<table>
<colgroup>
<col style="width: 16%" />
<col style="width: 45%" />
<col style="width: 18%" />
<col style="width: 18%" />
</colgroup>
<thead>
<tr class="header">
<th>Time</th>
<th>Speaker</th>
<th>Focus</th>
<th>Recording</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>8:00-8:30am</td>
<td>Nick Radcliffe and Pei Shan Yu, <em>Global Open Finance Centre of Excellence &amp; University of Edinburgh</em></td>
<td>Practices</td>
<td><a href="https://youtu.be/pWEc8XoIIKE" class="uri">https://youtu.be/pWEc8XoIIKE</a></td>
</tr>
<tr class="even">
<td>8:30-9:00am</td>
<td>Julia Schulte-Cloos, <em>LMU Munich</em></td>
<td>Practices</td>
<td>-</td>
</tr>
<tr class="odd">
<td>9:00-9:25am</td>
<td>Simeon Carstens, <em>Tweag/IO</em></td>
<td>Practices</td>
<td><a href="https://youtu.be/fpoFzDvrJAA" class="uri">https://youtu.be/fpoFzDvrJAA</a></td>
</tr>
<tr class="even">
<td>9:30-9:55am</td>
<td><em>Break</em></td>
<td>-</td>
<td>-</td>
</tr>
<tr class="odd">
<td>10:00-10:55am</td>
<td>Eva Vivalt, <em>University of Toronto</em></td>
<td>Keynote - Practices</td>
<td><a href="https://youtu.be/0WZUzf2oSGY" class="uri">https://youtu.be/0WZUzf2oSGY</a></td>
</tr>
<tr class="even">
<td>11:00-11:25am</td>
<td>Andrés Cruz, <em>Pontificia Universidad Católica de Chile</em></td>
<td>Practices</td>
<td><a href="https://youtu.be/HjdPDEACxmA" class="uri">https://youtu.be/HjdPDEACxmA</a></td>
</tr>
<tr class="odd">
<td>11:30-11:55am</td>
<td>Emily Riederer, <em>Capital One</em></td>
<td>Practices</td>
<td><a href="https://youtu.be/BknQ0ZNkMNY" class="uri">https://youtu.be/BknQ0ZNkMNY</a></td>
</tr>
<tr class="even">
<td>Noon-12:25pm</td>
<td>Florencia D’Andrea, <em>National Institute of Agricultural Technology</em></td>
<td>Practices</td>
<td><a href="https://youtu.be/9FVUIPfBeXw" class="uri">https://youtu.be/9FVUIPfBeXw</a></td>
</tr>
<tr class="odd">
<td>12:30-12:55pm</td>
<td>John Blischak, <em>Freelance scientific software developer</em></td>
<td>Practices</td>
<td><a href="https://youtu.be/RrcaGukYDyE" class="uri">https://youtu.be/RrcaGukYDyE</a></td>
</tr>
<tr class="even">
<td>1:00-1:25pm</td>
<td>Shemra Rizzo, <em>Genentech</em></td>
<td>Practices</td>
<td><a href="https://youtu.be/rEYtB3CG76Q" class="uri">https://youtu.be/rEYtB3CG76Q</a></td>
</tr>
<tr class="odd">
<td>1:30-2:25pm</td>
<td><em>Break</em></td>
<td>-</td>
<td>-</td>
</tr>
<tr class="even">
<td>2:30-2:55pm</td>
<td>Wijdan Tariq, <em>University of Toronto</em></td>
<td>Evaluating</td>
<td>-</td>
</tr>
<tr class="odd">
<td>3:00-3:25pm</td>
<td>Sharla Gelfand, <em>Freelance R Developer</em></td>
<td>Practices</td>
<td><a href="https://youtu.be/G5Nm-GpmrLw" class="uri">https://youtu.be/G5Nm-GpmrLw</a></td>
</tr>
<tr class="even">
<td>3:30-3:55pm</td>
<td>Ryan Briggs, <em>University of Guelph</em></td>
<td>Practices</td>
<td><a href="https://youtu.be/_dgGbxItiB4" class="uri">https://youtu.be/_dgGbxItiB4</a></td>
</tr>
<tr class="odd">
<td>4:00-4:25pm</td>
<td>Monica Alexander, <em>University of Toronto</em></td>
<td>Practices</td>
<td><a href="https://youtu.be/yvM2C6aZ94k" class="uri">https://youtu.be/yvM2C6aZ94k</a></td>
</tr>
<tr class="even">
<td>4:30-4:55pm</td>
<td>Annie Collins, <em>University of Toronto</em></td>
<td>Practices</td>
<td><a href="https://youtu.be/u4ibhN_nWyI" class="uri">https://youtu.be/u4ibhN_nWyI</a></td>
</tr>
<tr class="odd">
<td>5:00-5:25pm</td>
<td>Nancy Reid, <em>University of Toronto</em></td>
<td>Practices</td>
<td><a href="https://youtu.be/sIsOPuZOQL4" class="uri">https://youtu.be/sIsOPuZOQL4</a></td>
</tr>
<tr class="even">
<td>5:30-6:00pm</td>
<td>Rohan Alexander, <em>University of Toronto</em></td>
<td>Closing remarks</td>
<td><a href="https://youtu.be/7LttFNOI6p8" class="uri">https://youtu.be/7LttFNOI6p8</a></td>
</tr>
</tbody>
</table>
<h3 id="presenter-biographies-and-abstracts-1">Presenter biographies and abstracts</h3>
<p>Keynotes:</p>
<ul>
<li><a href="http://evavivalt.com/">Eva Vivalt</a>
<ul>
<li>Bio: Eva Vivalt is an Assistant Professor in the Department of Economics at the University of Toronto. Her main research interests are in cash transfers, reducing barriers to evidence-based decision-making, and global priorities research.</li>
<li>Abstract: An overview of the role of forecasting and a new platform for making them.</li>
</ul></li>
<li><a href="https://www2.stat.duke.edu/~mc301/">Mine Çetinkaya-Rundel</a>
<ul>
<li>Bio: Mine Çetinkaya-Rundel is a Senior Lecturer in Statistics and Data Science in the School of Maths at University of Edinburgh, and currently on leave as Associate Professor of the Practice in the Department of Statistical Science at Duke University as well as a Professional Educator and Data Scientist at RStudio. She is the author of three open source statistics textbooks and is an instructor for Coursera. She is the chair-elect of the Statistical Education Section of the American Statistical Association. Her work focuses on innovation in statistics pedagogy, with an emphasis on student-centered learning, computation, reproducible research, and open-source education.<br />
</li>
<li>Abstract: In the beginning was R Markdown. In this talk I will give a brief review of teaching statistics and data analysis through the lens of reproducibility with R Markdown, and how to use this tool effectively in teaching to maintain reproducibility as the scope of your students’ projects and their experience grow.</li>
</ul></li>
<li><a href="https://www.eva.mpg.de/ecology/staff/riana-minocher.html">Riana Minocher</a>
<ul>
<li>Bio: Riana Minocher is a doctoral student at the Max Planck Institute for Evolutionary Anthropology in Leipzig. She is an evolutionary biologist with broad interests. She has worked on a range of projects on human and non-human primate behaviour and ecology. She is particularly interested in the evolutionary processes that create and shape diversity between and within groups. Through her PhD research, she is keen on exploring the dynamics of cultural transmission and learning in human populations, to better understand the diverse patterns of behaviour we observe.<br />
</li>
<li>Abstract: Interest in improving reproducibility, replicability and transparency of research has increased substantially across scientific fields over the last few decades. We surveyed 560 empirical, quantitative publications published between 1955 and 2018, to estimate the rate of reproducibility for research on social learning, a large subfield of behavioural ecology. We found supporting materials were available for less than 30% of publications during this period. The availability of data declines exponentially with time since publication, with a half-life of about six years, and this “data decay rate” varies systematically with both study design and study species. Conditional on materials being available, we estimate that a reasonable researcher could expect to successfully reproduce about 80% of published results, based on our evaluating a subset of 40 publications. Taken together, this indicates an overall success rate of 24% for both acquiring materials and recovering published results, with non-reproducibility of results primarily due to unavailable, incomplete, or poorly-documented data. We provide recommendations to improve the reproducibility of research on the ecology and evolution of social behaviour.</li>
</ul></li>
</ul>
<p>Invited presentations:</p>
<ul>
<li><a href="http://www2.cs.queensu.ca/people/profile.php?fname=Amber&amp;lname=Simpson">Amber Simpson</a>
<ul>
<li>Bio: Amber Simpson is the Canada Research Chair in Biomedical Computing and Informatics and Associate Professor in the School of Computing (Faculty of Arts and Science) and Department of Biomedical and Molecular Sciences (Faculty of Health Sciences). She specializes in biomedical data science and computer-aided surgery. Her research group is focused on developing novel computational strategies for improving human health. She joined the Queen’s University faculty in 2019, after four years as faculty at Memorial Sloan Kettering Cancer Center in New York and three years as a Research Assistant Professor in Biomedical Engineering at Vanderbilt University in Nashville. She is an American Association of Cancer Research award winner and the holder of multiple National Institutes of Health grants. She received her PhD in Computer Science from Queen’s University.<br />
</li>
<li>Abstract: The development of predictive and prognostic biomarkers is a major area of investigation in cancer research. Our lab specializes in the development of quantitative imaging markers for personalized treatment of cancer. Progress in developing these novel markers is limited by a lack of optimization, standardization, and validation, all critical barriers to clinical use. This talk will describe our work in the repeatability and reproducibility of imaging biomarkers.</li>
</ul></li>
<li><a href="https://arcruz0.github.io/">Andrés Cruz</a>
<ul>
<li>Bio: Andrés Cruz is an adjunct instructor at Pontificia Universidad Católica de Chile, where he teaches computational social science. He holds a BA and MA in Political Science, and is the co-editor of “R for Political Data Science: A Practical Guide” (CRC Press, 2020), an R manual for social science students and practitioners.<br />
</li>
<li>Abstract: <code>inexact</code> is an RStudio addin to supervise fuzzy joins. Merging data sets is a simple procedure in most statistical software packages. However, applied researchers frequently face problems when dealing with data in which ID variables are not properly standardized. For instance, politicians’ names can be spelled differently in multiple sources (press reports, official documents, etc.), causing regular merging methods to fail. The most common approach to fix this issue when working with small and medium data sets is manually fixing the problematic values before merging. However, this solution is time-consuming and not reproducible. An RStudio addin called “inexact” was created to help with this. The package draws from approximate string matching algorithms, which quantify the distance between two given strings. When merging data sets with non-standardized ID variables, <code>inexact</code> users benefit from automatic match suggestions, while also being able to override the automatic choices when needed, using a user-friendly graphical user interface (GUI). The output is simply code to perform the corrected merging procedure, which registers the employed algorithm and any corrections made by the user, ensuring reproducibility. A development version of <code>inexact</code> is available on <a href="https://github.com/arcruz0/inexact">GitHub</a>.</li>
</ul></li>
<li>Annie Collins
<ul>
<li>Bio: Annie Collins is an undergraduate student in the Department of Mathematics specializing in applied mathematics and statistics with a minor in history and philosophy of science. In her free time, she focusses her efforts on student governance, promoting women’s representation in STEM, and working with data in the non-profit and charitable sector.<br />
</li>
<li>Abstract: We create a dataset of all the pre-prints published on medRxiv between 28 January 2020 and 31 January 2021. We extract the text from these pre-prints and parse them looking for keyword markers signalling the availability of the data and code underpinning the pre-print. We are unable to find markers of either open data or open code for 81 per cent of the pre-prints in our sample. Our paper demonstrates the need to have authors categorize the degree of openness of their pre-print as part of the medRxiv submissions process, and more broadly, the need to better integrate open science training into a wide range of fields</li>
</ul></li>
<li><a href="https://emilyriederer.netlify.app">Emily Riederer</a>
<ul>
<li>Bio: Emily Riederer is a Senior Analytics Manager at Capital One. Her team focuses on reimagining our analytical infrastructure by building data products, elevating business analysis with novel data sources and statistical methods, and providing consultation and training to our partner teams.<br />
</li>
<li>Abstract: Complex software systems make performance guarantees through documentation and unit tests, and they communicate these to users with conscientious interface design. However, published data tables exist in a gray area; they are static enough not to be considered a ‘service’ or ‘software’, yet too raw to earn attentive user interface design. This ambiguity creates a disconnect between data producers and consumers and poses a risk for analytical correctness and reproducibility. In this talk, I will explain how controlled vocabularies can be used to form contracts between data producers and data consumers. Explicitly embedding meaning in each component of variable names is a low-tech and low-friction approach which builds a shared understanding of how each field in the dataset is intended to work. Doing so can offload the burden of data producers by facilitating automated data validation and metadata management. At the same time, data consumers benefit by a reduction in the cognitive load to remember names, a deeper understanding of variable encoding, and opportunities to more efficiently analyze the resulting dataset. After discussing the theory of controlled vocabulary column-naming and related workflows, I will illustrate these ideas with a demonstration of the <code>convo</code> R package, which aids in the creation, upkeep, and application of controlled vocabularies. This talk is based on my related <a href="https://emilyriederer.netlify.app/post/column-name-contracts/">blog post</a> and <a href="https://emilyriederer.github.io/convo">R package</a>.</li>
</ul></li>
<li><a href="https://florencia.netlify.app/">Florencia D’Andrea</a>
<ul>
<li>Bio: Florencia D’Andrea is a post-doc at the Argentine National Institute of Agricultural Technology where she develops computer tools to assess the risk of pesticide applications for aquatic ecosystems. She holds a PhD in Biological Sciences from the University of Buenos Aires, Argentina, and is part of the ReproHack core-team and the R-Ladies global team. She believes that code and data should also be recognized as valuable products of scientific work.<br />
</li>
<li>Abstract: Choose your own adventure to a reproducible scientific article: learnings from ReproHack
“I shared the code and data of my last scientific article, does it mean that it is reproducible?”
One might think that having access to the research data and the code used to analyze that data would be enough to reproduce published results, but often this is much more involved. Is reproducibility dependent on the reviewer’s knowledge? What things do we not usually think about can affect reproducibility? Can the choice of how to capture the computational environment influence the experience of the reviewer? In this talk, we are going to think together some of the necessary steps that make someone else able to reproduce a scientific article or project. I will share some thoughts from my experience in ReproHack and show you how reviewing is a great practice to learn about reproducibility. What is ReproHack? Reprohack is a hackathon-style event focused on the reproducibility of research results. These hackathons provide a low-pressure sandbox environment for practicing reproducible research: Authors can practice producing reproducible research and receive friendly feedback and appreciation of their efforts. Participants can practice reviewing, learn about reproducibility best practices as well as common pitfalls from working with real-life materials rather than just dummy. They also get inspired and grow confidence in working more openly themselves. Research Community benefits from: Evaluating what best practice is in practice. More practice in both developing and reviewing materials.</li>
</ul></li>
<li><a href="https://www.ocf.berkeley.edu/~garret/">Garret Christensen</a>
<ul>
<li>Bio: Garret Christensen received his economics PhD from UC Berkeley in 2011. He is an economist with the FDIC. Before that he worked for the Census Bureau, and he was a project scientist with the Berkeley Initiative for Transparency in the Social Sciences and a Data Science Fellow with the Berkeley Institute for Data Science.<br />
</li>
<li>Abstract: Adoption of Open Science Practices is Increasing: Survey Evidence on Attitudes, Norms and Behavior in the Social Sciences. Has there been meaningful movement toward open science practices within the social sciences in recent years? Discussions about changes in practices such as posting data and pre-registering analyses have been marked by controversy—including controversy over the extent to which change has taken place. This study, based on the State of Social Science (3S) Survey, provides the first comprehensive assessment of awareness of, attitudes towards, perceived norms regarding, and adoption of open science practices within a broadly representative sample of scholars from four major social science disciplines: economics, political science, psychology, and sociology. We observe a steep increase in adoption: as of 2017, over 80% of scholars had used at least one such practice, rising from one quarter a decade earlier. Attitudes toward research transparency are on average similar between older and younger scholars, but the pace of change differs by field and methodology. According with theories of normal science and scientific change, the timing of increases in adoption coincides with technological innovations and institutional policies. Patterns are consistent with most scholars underestimating the trend toward open science in their discipline.</li>
</ul></li>
<li><a href="http://www.jakebowers.org">Jake Bowers</a>
<ul>
<li>Bio: Jake Bowers is a Senior Scientist at The Policy Lab and a member of the Lab’s data science practice. Jake is Associate Professor of Political Science and Statistics at the University of Illinois Urbana-Champaign. He has served as a Fellow in the Office of Evaluation Sciences in the General Services Administration of the US Federal Government and is Methods Director for the Evidence in Governance and Politics network. Jake holds a PhD in Political Science from the University of California, Berkeley, and a BA in Ethics, Politics and Economics from Yale University.<br />
</li>
<li>Abstract: For evidence-based public policy to grow in impact and importance, practices to enhance scientific credibility should be brought into governmental contexts and also should be modified for those contexts. For example, few analyses of governmental data allow data sharing (in contrast with most scientific studies); and many analyses of governmental administrative data inform high stakes immediate decisions (in contrast with the slow accumulation of scientific knowledge). We make several proposals to adjust scientific norms of reproducibility and pre-registration to the policy context.</li>
</ul></li>
<li><a href="https://jdblischak.com/">John Blischak</a>
<ul>
<li>Bio: John Blischak is a freelance scientific software developer for the life sciences industry. He is the primary author of the R package workflowr and the co-maintainer of the CRAN Task View on Reproducible Research. He received his PhD in Genetics from the University of Chicago.<br />
</li>
<li>Abstract: The <code>workflowr</code> R package helps organize computational research in a way that promotes effective project management, reproducibility, collaboration, and sharing of results. <code>workflowr</code> combines literate programming (knitr and rmarkdown) and version control (Git, via git2r) to generate a website containing time-stamped, versioned, and documented results. Any R user can quickly and easily adopt <code>workflowr</code>, which includes four key features: (1) <code>workflowr</code> automatically creates a directory structure for organizing data, code, and results; (2) <code>workflowr</code> uses the version control system Git to track different versions of the code and results without the user needing to understand Git syntax; (3) to support reproducibility, <code>workflowr</code> automatically includes code version information in webpages displaying results and; (4) <code>workflowr</code> facilitates online Web hosting (e.g. GitHub Pages) to share results. Our goal is that <code>workflowr</code> will make it easier for researchers to organize and communicate reproducible results. Documentation and source code are <a href="https://github.com/jdblischak/workflowr">available</a>.</li>
</ul></li>
<li><a href="https://jschultecloos.github.io">Julia Schulte-Cloos</a>
<ul>
<li>Bio: Julia Schulte-Cloos is a Marie Skłodowska-Curie funded research fellow at LMU Munich. She has earned her PhD in Political Science from the European University Institute. Julia is passionate about developing tools and templates for generating reproducible workflows and creating reproducible research outputs with R Markdown.<br />
</li>
<li>Abstract: We present a template package in R that allows users without any prior knowledge of R Markdown to implement reproducible research practices in their scientific workflows. We provide a single Rmd-file that is fully optimized for two different output formats, HTML and PDF. While in the stage of explorative analysis and when focusing on content only, researchers may rely on the ‘draft mode’ of our template that knits to HTML When in the stage of research dissemination and when focusing on the presentation of results, in contrast, researchers may rely on the ‘manuscript mode’ that knits to PDF. Our template outlines the basics for successfully writing a reproducible paper in R Markdown by showing how to include citations, figures, and cross-references. It also provides examples for the use of <code>ggplot2</code> to include plots, both in static and animated outputs, and it shows how to present the most commonly used tables in scientific research (descriptive statistics and regression tables). Finally, in our template, we discuss some more advanced features of literate programming and helpful tweaks in R Markdown.</li>
</ul></li>
<li><a href="https://jazzystats.com/">Lauren Kennedy</a>
<ul>
<li>Bio: Lauren Kennedy is a lecturer in the Econometrics and Business Statistics department at Monash University. She works on applied statistical problems in the social sciences using primarily Bayesian methodology. Her most recent work is with survey data, particularly the use of model and poststratify methods to make population and subpopulation predictions.<br />
</li>
<li>Abstract: Survey data is challenging to work with. It frequently contains entry errors (either from respondent recollection or interviewer entry) that are difficult to verify and identify. Survey data is often received in a form that is sensible for the software for which entry is intuitive, which does not necessarily follow through to a data structure that is intuitive to work with as an analyst. When we consider the use of tools like multilevel regression and poststratification, our challenges compound. Even if the population data is precleaned before release, measurements and items in the sample need to be mapped to measurements and items in the population. In this talk we discuss case studies of how and where these challenges appear in practice.</li>
</ul></li>
<li><a href="https://apnews.com/article/1382560004">Larry Fenn</a>
<ul>
<li>Bio: Larry Fenn is a data journalist at the Associated Press. His investigative work has covered a broad range of topics, from guns to education to housing policy. Prior to journalism, he was an adjunct lecturer at Hunter College for applied mathematics and statistics.<br />
</li>
<li>Abstract: Please see Meghan Hoyer.</li>
</ul></li>
<li><a href="https://pacha.dev">Mauricio Vargas Sepúlveda</a>
<ul>
<li>Bio: Mauricio Vargas Sepúlveda loves working with data and statistical programming, and is constantly learning new skills and tooling in his spare time. He mostly works in R due to its huge number of libraries and emphasis on reproducible analysis.<br />
</li>
<li>Abstract: Evidence-based policymaking has turned into a high priority for governments across the world. The possibility of gaining efficiencies in the public expenditure and linking the policy design to the desired outcomes have been presented as significant advantages for the field of comparative policy. However, the same movement that supports the use of evidence in public policy decision making has brought a great concern about the sources of the supposed evidence. How should policymakers evaluate the evidence? The possibilities are open and depend on the institutional arrangements that support governmental operation and the possibility of properly judging the nature of the evidence. The movement of science reproducibility could enlighten the discussion about the quality of the evidence by providing a structured approach towards the source’s validity based on the possibility of reproducing the logic and analysis proper of scientific communication. This paper attempts to analyze the nature and quality of civil society organizations’ contributions to develop evidence for policymaking process from reproducibility perspective.<br />
</li>
</ul></li>
<li><a href="https://twitter.com/meghanhoyer">Meghan Hoyer</a>
<ul>
<li>Bio: Meghan Hoyer is Data Director at The Washington Post where she leads data projects and acts as a consulting editor on data-driven stories, graphics and visualizations across the newsroom. Before this she helped lead the AP’s data journalism. Meghan earned a bachelor of science in journalism at Northwestern University and an MFA in creative nonfiction writing at Old Dominion University.<br />
</li>
<li>Abstract: This talk will cover AP DataKit, which is an open-source command-line tool designed to better structure and manage projects, and more generally, talk about creating sane, reproducible workflows.</li>
</ul></li>
<li><a href="https://www.monicaalexander.com/">Monica Alexander</a>
<ul>
<li>Bio: Monica Alexander is an Assistant Professor in Statistical Sciences and Sociology at the University of Toronto. She received her PhD in Demography from the University of California, Berkeley. Her research interests include statistical demography, mortality and health inequalities, and computational social science.<br />
</li>
<li>Abstract: Sharing code for papers and projects is an important part of reproducible research. However, sometimes sharing code may be difficult, if the researcher feels their code is ‘not good enough’ and may reflect poorly on their broader research skills. This presentation contains some brief reflections from research, consulting, and teaching experiences that have led to overcoming my own barriers to sharing code and to help others do the same.<br />
</li>
</ul></li>
<li><a href="http://www.utstat.utoronto.ca/reid/">Nancy Reid</a>
<ul>
<li>Bio: Nancy Reid is Professor of Statistical Sciences at the University of Toronto and Canada Research Chair in Statistical Theory and Applications. Her main area of research is theoretical statistics. This treats the foundations and properties of methods of statistical inference. She is interested in how best to use information in the data to construct inferential statements about quantities of interest. A very simple example of this is the widely quoted ‘margin of error’ in the reporting of polls, another is the ubiquitous ‘p-value’ reported in medical and health studies. Much of her research considers how to ensure that these inferential statements are both accurate and effective at summarizing complex sets of data.<br />
</li>
<li>Abstract: Are p-values contributing to a crisis in replicability and reproducibility? This has been the topic of many dialogues, diatribes, and discussions among statisticians and scientists in recent years. I will share my thoughts on the issues, with emphasis on the role of inferential theory in helping to clarify the arguments.</li>
</ul></li>
<li><a href="https://www.stochasticsolutions.com/about/">Nick Radcliffe</a>
<ul>
<li>Bio: Nick Radcliffe is the founder of the data science consulting and software firm, Stochastic Solutions Limited, the Interim Chief Scientist at the Global Open Finance Centre of Excellence, and a Visiting Professor in Maths and Stats at University of Edinburgh, Scotland. His background combines theoretical physics, operations research, machine learning and stochastic optimization. Nick’s current research interests include a focus on test-driven data analysis, (an approach to improving correctness of analytical results that combines ideas from reproducible research and test-driven development) and privacy-respecting analysis. He is the lead author of the open-source Python tdda package, which provides practical tools for testing analytical software and data, and also of the Miró data analysis suite.<br />
</li>
<li>Abstract: The Global Open Finance Centre of Excellence is currently engaged in analysis of the financial impact of COVID-19 on the citizens and businesses of the UK. This research uses non-consented but de-identified financial data on individuals and businesses, on the basis of legitimate interest. All analysis is carried out in a highly locked-down analytical environment known as a Safe Haven. This talk will explain our approach to the challenges of ensuring the correctness and robustness of results in an environment where neither code nor input data can be opened up for review and even outputs need to be subject to disclosure control to reduce further any risks to privacy. Topics will include: testing input data for conformance and lack of personal identifiers using constraints; multiple implementations and verification of equivalence of results; regression tests and reference tests; verification of output artefacts; verification of output disclosure controls; data provenance and audit trails; test-driven data analysis—the underlying philosophy (and library) that we use to underpin this work.</li>
</ul></li>
<li><a href="https://cord.asu.edu/content/nicolas-didier">Nicolas Didier</a>
<ul>
<li>Bio: Nicolas Didier is studying for a PhD in Public Administration and Policy at the Arizona State University. During his PhD studies and previous studies, he has worked extensively on developing evidence that addresses policy in labour markets and public expenditure.<br />
</li>
<li>Abstract: Please see Mauricio Vargas Sepúlveda.</li>
</ul></li>
<li><a href="https://www.ryancbriggs.net">Ryan Briggs</a>
<ul>
<li>Bio: Ryan Briggs is a social scientist who studies the political economy of poverty alleviation. Most of his research focuses on the spatial targeting of foreign aid. He is an Assistant Professor in the Guelph Institute of Development Studies and Department of Political Science at the University of Guelph. Before that, he taught at Virginia Tech and American University.<br />
</li>
<li>Abstract: It is hard to do research. One reason for this is that it has a production function where one low quality input (among many high quality inputs) can poison a final result. This talk explains how such ‘o-ring’ production functions work and draws out lessons for applied researchers.</li>
</ul></li>
<li><a href="https://sharla.party/">Sharla Gelfand</a>
<ul>
<li>Bio: Sharla Gelfand is a freelance R and Shiny developer specializing in enabling easy access to data and replacing manual, redundant processes with ones that are automated, reproducible, and repeatable. They also co-organize R-Ladies Toronto and the Greater Toronto Area R User Group. They like R (of course), dogs, learning Spanish, playing bass, and punk.<br />
</li>
<li>Abstract: Getting stuck, looking around for a solution, and eventually asking for help is an inevitable and constant aspect of being a programmer. If you’ve ever looked up a question only to find some brave soul getting torn apart on Stack Overflow for not providing a minimum working example, you know it’s also one of the most intimidating parts! A minimum working example, or a reproducible example as it’s more often called in the R world, is one of the best ways to get help with your code - but what exactly is a reproducible example? How do you create one, and do it efficiently? Why is it so scary? This talk will cover what components are needed to make a good reproducible example to maximize your ability to get help (and to help yourself!), strategies for coming up with an example and testing its reproducibility, and why you should care about making one. We will also discuss how to extend the concept of reproducible examples beyond “Help! my code doesn’t work” to other environments where you might want to share code, like teaching and blogging.</li>
</ul></li>
<li><a href="https://www.linkedin.com/in/shemra">Shemra Rizzo</a>
<ul>
<li>Bio: Shemra Rizzo is a senior data scientist in Genentech’s Personalized Healthcare group. Shemra’s role includes research on COVID-19 using electronic health records, and the development of data-driven approaches to evaluate clinical trial eligibility criteria. Shemra obtained her PhD in Biostatistics from UCLA. Before joining Genentech, Shemra was an assistant professor of statistics at UC Riverside, where her research covered topics in mental health, health disparities, and nutrition. In her free time, Shemra enjoys spending time with her family and running.<br />
</li>
<li>Abstract: Real world data for an emerging disease has unique challenges. In this talk, I’ll describe how our group made sense of complex Electronic Health Records (EHR) data for COVID19 early in the pandemic. I will share our experience working towards reliable, replicable and reproducible studies using EHR licensed data.</li>
</ul></li>
<li><a href="https://www.shirokuriwaki.com/">Shiro Kuriwaki</a>
<ul>
<li>Bio: Shiro Kuriwaki is a PhD Candidate in the Department of Government at Harvard University. His research focuses on democratic representation in American Politics. In an ongoing project, he studies the structure of voter’s choices across levels of government and the political economy of local elections, using cast vote records and surveys. His other projects also help understand the mechanics of representation, including: public opinion and Congress, modern survey statistics and causal inference, and election administration. Prior to and during graduate school, he worked at the Analyst Institute in Washington D.C.<br />
</li>
<li>Abstract: I show how new features of the <a href="https://github.com/IQSS/dataverse-client-r">dataverse</a> R package facilitate reproducibility in empirical, substantive projects. While packages and scripts make our code transparent and portable across forms, the import of large and complex datasets is often a nuisance in project workflows that involve various data cleaning and wrangling tasks. And the GUI for Dataverse can be sometimes tedious to integrate into code-based workflow. Will Beasley and I, along with multiple other contributors, updated the dataverse R package for the first time since 2017 with the goal of spreading its use in empirical workflow. In this iteration, we make it easier to retrieve dataframes of various file format and options for version specification and variable subsetting. I also discuss the latest updates to pyDataverse, a independent implementation in Python which is currently more advanced in its implementation but focused on uploading and creating datasets to dataverse.</li>
</ul></li>
<li><a href="http://simeon-carstens.com/">Simeon Carstens</a>
<ul>
<li>Bio: Simeon Carstens is a Data Scientist at Tweag I/O, a software innovation lab and consulting company. Originally a physicist, Simeon did a PhD and postdoc research in computational biology, focusing on Bayesian determination of three-dimensional chromosome structures.<br />
</li>
<li>Abstract: Data analysis often requires a complex software environment containing one or several programming languages, language-specific modules and external dependencies, all in compatible versions. This poses a challenge to reproducibility: what good is a well-designed, tested and documented data analysis pipeline if it is difficult to replicate the software environment required to run it? Standard tools such as Python / R virtual environments solve part of the problem, but do not take into account external and system-level dependencies. Nix is a fully declarative, open-source package manager solving this problem: a program packaged with Nix comes with a complete description of its full dependency tree, down to system libraries. In this presentation, I will give an introduction to Nix, show in a live demo how to set up a fully reproducible software environment and compare Nix to existing solutions such as virtual environments and Docker.</li>
</ul></li>
<li><a href="https://www.tiffanytimbers.com/">Tiffany Timbers</a>
<ul>
<li>Bio: Tiffany Timbers is an Assistant Professor of Teaching in the Department of Statistics and an Co-Director for the Master of Data Science program (Vancouver Option) at the University of British Columbia. In these roles she teaches and develops curriculum around the responsible application of Data Science to solve real-world problems. One of her favourite courses she teaches is a graduate course on collaborative software development, which focuses on teaching how to create R and Python packages using modern tools and workflows.<br />
</li>
<li>Abstract: In the data science courses at UBC, we define data science as the study and development of reproducible and auditable processes to obtain value (i.e., insight) from data. While reproducibility is core to our definition, most data science learners enter the field with other aspects of data science in mind, such as predictive modelling. This fact, along with the highly technical nature of the industry standard reproducibility tools currently employed in data science, present out-of-the gate challenges in teaching reproducibility in the data science classroom. Put simply, students are not as intrinsically motivated to learn this topic, and it is not an easy one for them to learn. What can a data science educator do? Over several iterations of teaching courses focused on reproducible data science tools and workflows, we have found that motivation, direct instruction and practice are key to effectively teach this challenging, yet important subject. In this talk, I will present examples of how we deeply motivate, effectively instruct and provide ample practice opportunities to our Master of Data Science students to effectively engage them in learning about this topic.</li>
</ul></li>
<li><a href="https://tombarton.netlify.app">Tom Barton</a>
<ul>
<li>Bio: Tom Barton is a PhD student in Politics at Royal Holloway, University of London. His PhD focuses on the impact of Voter Identification laws on political participation and attitudes. More generally his interests include elections, public opinion (particularly social values) and quantitative research methods.<br />
</li>
<li>Abstract: I reproduce Surridge, 2016, ‘Education and liberalism: pursuing the link’, <em>Oxford Review of Education</em>, 42:2, pp. 146-164, using the 1970 British Cohort Study (BCS70), instead using a difference-in-difference regression approach with more waves of data. I find that whilst there is evidence for both socialisation and self-selection models, self-selection dominates the link between social values and university attendance. This is counter to what Surridge (2016) concluded. The need for re-specification was two-fold, first Surridge’s methodology did not fully test for causality and secondly later waves have data have become available since.<br />
</li>
</ul></li>
<li><a href="https://www.tylergirard.com/">Tyler Girard</a>
<ul>
<li>Bio: Tyler Girard is a PhD Candidate in political science at the University of Western Ontario (London, Ontario, Canada). His dissertation research seeks to explain the origins and diffusion of the global financial inclusion agenda by focusing on the role of ambiguous ideas in mobilizing and consolidating transnational coalitions. More generally, his work also explores new approaches to conceptual measurement in international relations.<br />
</li>
<li>Abstract: In what ways can we incorporate reproducible practices in pedagogy for social science courses? I discuss how individual and group exercises centered around the replication of existing datasets and analyses offer a flexible tool for experiential learning. However, maximizing the benefits of such an approach requires customizing the activity to the students and the availability of instructor support. I offer several suggestions for effectively using replication exercises in both undergraduate and graduate level courses.</li>
</ul></li>
<li><a href="http://www.wijdantariq.com/">Wijdan Tariq</a>
<ul>
<li>Bio: Wijdan Tariq is an undergraduate student in the Department of Statistical Sciences at the University of Toronto.<br />
</li>
<li>Abstract: I undertake a narrow replication of Caicedo, 2019, ‘The Mission: Human Capital Transmission, Economic Persistence, and Culture in South America’, <em>Quarterly Journal of Economics</em>, 134:1, pp. 507-556. Caicedo reports of a remarkable, religiously inspired human capital intervention that took place in remote parts of South America 250 years ago and whose positive economic effects, he claims, persist to this day. I replicate some of the paper’s key results using data files that are available on the Harvard Dataverse portal. I discuss some lessons learned in the process of replicating this paper and share some reflections on the state of reproducibility in economics.</li>
</ul></li>
<li><a href="https://www.statistics.utoronto.ca/people/directories/graduate-students/yanbo-tang">Yanbo Tang</a>
<ul>
<li>Bio: Yanbo Tang is a PhD candidate at the University of Toronto in the Department of Statistical Sciences, under the joint supervision of Nancy Reid and Daniel Roy. He is interested in the study and application of methods in higher order asymptotics and statistical inference in the presence of many nuisance parameters. Nowadays, he works under the careful gaze of his pet parrot.<br />
</li>
<li>Abstract: Hypothesis testing results often rely on simple, yet important assumptions about the behavior of the distribution of p-values under the null and alternative. We show that commonly held beliefs regarding the distribution of p-values are misleading when the variance or location of the test statistic are not well-calibrated or when the higher order cumulants of the test statistic are not negligible. We further examine the impact of having these misleading p-values on reproducibility of scientific studies, with some examples focused on GWAS studies. Certain corrected tests are proposed and are shown to perform better than their traditional counterparts in certain settings.</li>
</ul></li>
</ul>
<h2 id="code-of-conduct">Code of conduct</h2>
<p><strong>Code</strong></p>
<p>The organizers of the Toronto Workshop on Reproducibility are dedicated to providing a harassment-free experience for everyone regardless of age, gender, sexual orientation, disability, physical appearance, race, or religion (or lack thereof).</p>
<p>All participants (including attendees, speakers, sponsors and volunteers) at the Toronto Workshop on Reproducibility are required to agree to the following code of conduct.</p>
<p>The code of conduct applies to all conference activities including talks, panels, workshops, and social events. It extends to conference-specific exchanges on social media, for instance posts tagged with the identifier of the conference (e.g. #TOrepro on Twitter), and replies to such posts.</p>
<p>Organizers will enforce this code throughout and expect cooperation in ensuring a safe environment for all.</p>
<p><strong>Expected Behaviour</strong></p>
<p>All conference participants agree to:</p>
<ul>
<li>Be considerate in language and actions, and respect the boundaries of fellow participants.</li>
<li>Refrain from demeaning, discriminatory, or harassing behaviour and language. Please refer to ‘Unacceptable Behaviour’ for more details.</li>
<li>Alert Rohan Alexander - <a href="mailto:rohan.alexander@utoronto.ca" class="email">rohan.alexander@utoronto.ca</a> - or Kelly Lyons - <a href="mailto:kelly.lyons@utoronto.ca" class="email">kelly.lyons@utoronto.ca</a> - if you notice someone in distress, or observe violations of this code of conduct, even if they seem inconsequential. Please refer to the section titled ‘What To Do If You Witness or Are Subject To Unacceptable Behaviour’ for more details.</li>
</ul>
<p><strong>Unacceptable Behaviour</strong></p>
<p>Behaviour that is unacceptable includes, but is not limited to:</p>
<ul>
<li>Stalking</li>
<li>Deliberate intimidation</li>
<li>Unwanted photography or recording</li>
<li>Sustained or willful disruption of talks or other events</li>
<li>Use of sexual or discriminatory imagery, comments, or jokes</li>
<li>Offensive comments related to age, gender, sexual orientation, disability, race or religion</li>
<li>Inappropriate physical contact, which can include grabbing, or massaging or hugging without consent.</li>
<li>Unwelcome sexual attention, which can include inappropriate questions of a sexual nature, asking for sexual favours or repeatedly asking for dates or contact information.</li>
</ul>
<p>If you are asked to stop harassing behaviour you should stop immediately, even if your behaviour was meant to be friendly or a joke, it was clearly not taken that way and for the comfort of all conference attendees you should stop.</p>
<p>Attendees who behave in a manner deemed inappropriate are subject to actions listed under ‘Procedure for Code of Conduct Violations’.</p>
<p><strong>Additional Requirements for Conference Contributions</strong></p>
<p>Presentation slides and posters should not contain offensive or sexualised material. If this material is impossible to avoid given the topic (for example text mining of material from hate sites) the existence of this material should be noted in the abstract and, in the case of oral contributions, at the start of the talk or session.</p>
<p><strong>Procedure for Code of Conduct Violations</strong></p>
<p>The organizing committee reserves the right to determine the appropriate response for all code of conduct violations. Potential responses include:</p>
<ul>
<li>a formal warning to stop harassing behaviour</li>
<li>expulsion from the conference</li>
<li>cancellation or early termination of talks or other contributions to the program</li>
</ul>
<p><strong>What To Do If You Witness or Are Subject To Unacceptable Behaviour</strong></p>
<p>If you are being harassed, notice that someone else is being harassed, or have any other concerns relating to harassment, please contact Rohan Alexander - <a href="mailto:rohan.alexander@utoronto.ca" class="email">rohan.alexander@utoronto.ca</a>, or Kelly Lyons - <a href="mailto:kelly.lyons@utoronto.ca" class="email">kelly.lyons@utoronto.ca</a>.</p>
<p>We will take all good-faith reports of harassment by Toronto Workshop on Reproducibility participants seriously.</p>
<p>We reserve the right to reject any report we believe to have been made in bad faith. This includes reports intended to silence legitimate criticism.</p>
<p>We will respect confidentiality requests for the purpose of protecting victims of abuse. We will not name harassment victims without their affirmative consent.</p>
<p>Questions or concerns about the Code of Conduct can be addressed to <a href="mailto:rohan.alexander@utoronto.ca" class="email">rohan.alexander@utoronto.ca</a>.</p>
<p><strong>Acknowledgements</strong></p>
<p>Parts of the above text are licensed CC BY-SA 4.0. Credit to SRCCON. This code of conduct was based on that developed for useR! 2018 which was a revision of the code of conduct used at previous useR!s and also drew from rOpenSci’s code of conduct.</p>
<!-- ```{r, out.width= "100%"} -->
<!-- knitr::include_graphics(c("images/reproducibility.png"))  -->
<!-- ``` -->
<div class="sourceCode" id="cb1"><pre class="sourceCode r distill-force-highlighting-css"><code class="sourceCode r"></code></pre></div>
<!--radix_placeholder_article_footer-->
<!--/radix_placeholder_article_footer-->
</div>

<div class="d-appendix">
</div>


<!--radix_placeholder_site_after_body-->
<!--/radix_placeholder_site_after_body-->
<!--radix_placeholder_appendices-->
<div class="appendix-bottom"></div>
<!--/radix_placeholder_appendices-->
<!--radix_placeholder_navigation_after_body-->
<!--/radix_placeholder_navigation_after_body-->

</body>

</html>
